<!DOCTYPE html><html lang="zh-Hans"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no"><meta name="author" content="Yansen"><meta name="renderer" content="webkit"><meta name="copyright" content="Yansen"><meta name="keywords" content="Hexo"><meta name="description" content=""><meta name="Cache-Control" content="no-cache"><meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"><title>Animal Genetic</title><link rel="stylesheet" href="/css/style.css?v=2018.7.9"><link rel="stylesheet" href="/css/animation.css?v=2018.7.9"><link rel="icon" href="/img/assets/favicon.ico"><link rel="stylesheet" href="https://cdn.bootcss.com/font-awesome/4.7.0/css/font-awesome.min.css?version=1.5.6"><!-- scripts--><script>(function( w ){
  "use strict";
  // rel=preload support test
  if( !w.loadCSS ){
    w.loadCSS = function(){};
  }
  // define on the loadCSS obj
  var rp = loadCSS.relpreload = {};
  // rel=preload feature support test
  // runs once and returns a function for compat purposes
  rp.support = (function(){
    var ret;
    try {
      ret = w.document.createElement( "link" ).relList.supports( "preload" );
    } catch (e) {
      ret = false;
    }
    return function(){
      return ret;
    };
  })();

  // if preload isn't supported, get an asynchronous load by using a non-matching media attribute
  // then change that media back to its intended value on load
  rp.bindMediaToggle = function( link ){
    // remember existing media attr for ultimate state, or default to 'all'
    var finalMedia = link.media || "all";

    function enableStylesheet(){
      link.media = finalMedia;
    }

    // bind load handlers to enable media
    if( link.addEventListener ){
      link.addEventListener( "load", enableStylesheet );
    } else if( link.attachEvent ){
      link.attachEvent( "onload", enableStylesheet );
    }

    // Set rel and non-applicable media type to start an async request
    // note: timeout allows this to happen async to let rendering continue in IE
    setTimeout(function(){
      link.rel = "stylesheet";
      link.media = "only x";
    });
    // also enable media after 3 seconds,
    // which will catch very old browsers (android 2.x, old firefox) that don't support onload on link
    setTimeout( enableStylesheet, 3000 );
  };

  // loop through link elements in DOM
  rp.poly = function(){
    // double check this to prevent external calls from running
    if( rp.support() ){
      return;
    }
    var links = w.document.getElementsByTagName( "link" );
    for( var i = 0; i < links.length; i++ ){
      var link = links[ i ];
      // qualify links to those with rel=preload and as=style attrs
      if( link.rel === "preload" && link.getAttribute( "as" ) === "style" && !link.getAttribute( "data-loadcss" ) ){
        // prevent rerunning on link
        link.setAttribute( "data-loadcss", true );
        // bind listeners to toggle media back
        rp.bindMediaToggle( link );
      }
    }
  };

  // if unsupported, run the polyfill
  if( !rp.support() ){
    // run once at least
    rp.poly();

    // rerun poly on an interval until onload
    var run = w.setInterval( rp.poly, 500 );
    if( w.addEventListener ){
      w.addEventListener( "load", function(){
        rp.poly();
        w.clearInterval( run );
      } );
    } else if( w.attachEvent ){
      w.attachEvent( "onload", function(){
        rp.poly();
        w.clearInterval( run );
      } );
    }
  }


  // commonjs
  if( typeof exports !== "undefined" ){
    exports.loadCSS = loadCSS;
  }
  else {
    w.loadCSS = loadCSS;
  }
}( typeof global !== "undefined" ? global : this ) );</script><script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js" defer></script><script src="/js/main.js?v=2018.7.9" defer></script><!-- fancybox--><link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'"><script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" defer></script><!-- busuanzi--><script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script><meta name="generator" content="Hexo 4.2.0"></head><body><section class="profile-close" id="cxo-profile"><div class="profile-avatar"><i class="fa fa-caret-left"></i><img src="img/assets/cat.png"></div><!--.profile-saying
  i.fa.fa-comment
  .saying--><div class="cxo-profile-inner"><div class="profile-name">Animal Genetic</div><div class="profile-signature">LOVE</div><div class="friends"><div>FRIENDS</div><span><a href="//github.com/Yansen0515" target="_black">friendA</a></span><span><a href="//github.com/" target="_black">friendB</a></span><span><a href="//github.com/" target="_black">friendC</a></span></div><div class="read-progress"></div></div></section><header id="cxo-intro" style="height: 70vh;background-image: url(/img/intro/index-bg.png);"><nav id="cxo-intro-nav"><section><div class="intro-nav-title"><a href="/" target="_blank" rel="noopener">Yan&amp;Qing</a></div><div class="intro-nav-label-box"><a href="/" target="_blank" rel="noopener">Home</a><a href="/about/" target="_blank" rel="noopener">About</a><a href="/archives/" target="_blank" rel="noopener">Archives</a><a href="/tags/" target="_blank" rel="noopener">Tags</a><a href="/Categories/" target="_blank" rel="noopener">Categories</a></div><i class="fa fa-bars intro-nav-menu"><div class="intro-nav-drop"><a href="/" target="_blank" rel="noopener">Home</a><a href="/about/" target="_blank" rel="noopener">About</a><a href="/archives/" target="_blank" rel="noopener">Archives</a><a href="/tags/" target="_blank" rel="noopener">Tags</a><a href="/Categories/" target="_blank" rel="noopener">Categories</a></div></i><div class="clear"></div></section></nav><h1>Yan&amp;Qing</h1><h5>Welcome to my Blog!</h5><div class="intro-social"><a class="fa fa-envelope" href="mailto:yansen0515@gmail" title="email"></a><a class="fa fa-github" href="//github.com/Yansen0515" target="_blank" title="github"></a><a class="fa fa-weibo" href="//www.weibo.com" target="_blank" title="weibo"></a><a class="fa fa-facebook" href="//www.facebook.com/Yansen Chen" target="_blank" title="facebook"></a><a class="fa fa-twitter" href="//twitter.com" target="_blank" title="twitter"></a><a class="fa fa-rss" href="/atom.xml" target="_blank" title="rss"></a></div></header><article class="cxo-up" id="cxo-content-outer"><section id="cxo-content-inner"><section id="recent-posts"><div class="recent-post-item"><div class="post-title"><a href="2020/05/25/GS/" target="_blank" rel="noopener">7 - Genomic Prediction（GP）/Genomic Selection (GS)</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-05-25T19:31:09.754Z">2020-05-25 21:31:09</time><i class="fa fa-edit"></i><time datetime="2020-05-25T19:31:09.754Z">2020-05-26 12:15:14</time></div><div class="recent-post-content"> GP中的一些算法首先大家要有一定基础知识：        首先，现在实际基因组选择都是使用的SNP        位点可以使用0 1 2表示，主要等位基因为0， 次要等位基因为1        表型一般都会做校正： Daughter Yield Deviations（ DYD， 奶牛使用较多）， Deregressed EBV（DRP）
GS分类    1 多步法        （1） 计算EBV        （2） 取出高可靠性的公牛EBV， 进行计算出DYD与DRP        （3） 估计SNP的effect        (4) 评估新个体的Direct Genomic Valu ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="遗传算法"> 遗传算法</a></div><div class="read-more"><a class="more" href="2020/05/25/GS/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/05/25/多性状动物模型/" target="_blank" rel="noopener">6 - MT-MME（多性状动物模型）</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-05-25T15:33:06.032Z">2020-05-25 17:33:06</time><i class="fa fa-edit"></i><time datetime="2020-05-25T15:33:06.032Z">2020-05-25 21:30:25</time></div><div class="recent-post-content">多性状动物模型主要用于以下情况：    1 同时具有几个性状    2 性状之间的相关性的使用        同一个动物： 弥补部分缺失值        相关动物：males &lt;=&gt; female    3 实际中特别感兴趣的性状如果：        具有明确生物关系： 最优相关 -/+ 0.5        研究缺失值        预测不知道的性状
计算复杂， 需要好的计算算法                         和  计算机            eg:Canonical transformation             MACE                ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="遗传算法"> 遗传算法</a></div><div class="read-more"><a class="more" href="2020/05/25/多性状动物模型/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/05/20/OLSandSI/" target="_blank" rel="noopener">1 - OLS and SI</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-05-20T20:28:23.874Z">2020-05-20 22:28:23</time><i class="fa fa-edit"></i><time datetime="2020-05-20T20:28:23.874Z">2020-05-25 20:46:55</time></div><div class="recent-post-content">前言： 对自己上半年的一个遗传课程“Genetics, Genomics and (Bio)diversity”整理，其是Liege university 的Nicolas Gengler主讲课程都是由Octave语言，完成练习。多使用矩阵计算，需要相关基础知识：如：Kronecker product; Cholsky分解； Cannoical分解； 广义逆，
###############first part OLS简单的线性回归模型：y = b1 + b2*xb1为截距，b2为斜率
矩阵写法： y = Wb + e    （1）
求解时： 我们需要使e最小此方法被称为： Least squ ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="遗传算法"> 遗传算法</a></div><div class="read-more"><a class="more" href="2020/05/20/OLSandSI/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/05/25/difference_animal_model/" target="_blank" rel="noopener">5 - different model of MME</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-05-25T13:26:27.933Z">2020-05-25 15:26:27</time><i class="fa fa-edit"></i><time datetime="2020-05-25T13:26:27.933Z">2020-05-25 17:32:16</time></div><div class="recent-post-content">先再次写下，MME的一般式子Mixed model equations(MME)
[X&apos;R-1X  X&apos;R-1Z      [β    [X&apos;R-1y
Z&apos;R-1X  Z&apos;R-1Z+G-1]  u] =  Z&apos;R-1y]       

or
简写  Cs = r求解时，需要求G或C的逆矩阵， 因计算量过大， 1960年以前，不能实现
所以人们就先简化了MME    1 简化 Var(u)    2 只使用males    3 简化关系矩阵        （1） 只使用sirs求        （2） 在Z中表示关系矩阵        ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="遗传算法"> 遗传算法</a></div><div class="read-more"><a class="more" href="2020/05/25/difference_animal_model/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/05/25/Amatrix/" target="_blank" rel="noopener">4 - A Matrix(additive relationship matrix)</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-05-25T09:21:29.851Z">2020-05-25 11:21:29</time><i class="fa fa-edit"></i><time datetime="2020-05-25T09:21:29.851Z">2020-05-25 15:26:09</time></div><div class="recent-post-content">在遗传分析中，个体之间的关系是非常重要的。这里先只介绍基于系谱关系的关系矩阵（A matrix）及其逆矩阵。以后再介绍G（GLUP使用）和H（SSBLUP使用）矩阵。
A矩阵现主要有两个定义观点：        1 根据近交系数（单个个体）： 在相同位点携带的两个alleles来自同一个祖先的概率        2 相关系数（两个个体之间）： 这两只动物的后代的近交系数的两倍个体近交系数：  0 ≤ Fx &lt; 1个体的相关系数： a = 1 + Fx两个个体之间的相关系数： a = 2Fx
计算方法：    1 路径系数法（Wright，1922）        axy = (2-n)1 ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="遗传算法"> 遗传算法</a></div><div class="read-more"><a class="more" href="2020/05/25/Amatrix/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/05/24/BLUP1/" target="_blank" rel="noopener">3 - BLUP的基本特性及求解方法</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-05-24T14:59:35.107Z">2020-05-24 16:59:35</time><i class="fa fa-edit"></i><time datetime="2020-05-24T14:59:35.107Z">2020-05-25 11:20:29</time></div><div class="recent-post-content">如MME中所讲， Henderson的想法是结合OLS与ISOLS解出的u太大。IS是对系数采用每个奶牛的女儿数进行了一定修改， 或其可以加上一个假设值D实际上均是对Z’（ZZ’ + Iλ ）-1 = (ZZ‘)-1Z’ + (Iλ)-1Z’ 中的λ修改而λ = (1 - h2)/h2  其为动物模型    如果h2 = 0, 则D为无限大        h2 = 1, D =0
 Sire model 为(4 - h2)/h2 
Penalized OLS estimator: u = (ZZ’ + D)-1Z’(y - Xβ)SI estimator: u = AZ’(ZAZ’ + Iλ) ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="遗传算法"> 遗传算法</a></div><div class="read-more"><a class="more" href="2020/05/24/BLUP1/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/05/23/MME/" target="_blank" rel="noopener">2 - MME - Mixed models</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-05-23T20:03:13.082Z">2020-05-23 22:03:13</time><i class="fa fa-edit"></i><time datetime="2020-05-23T20:03:13.082Z">2020-05-25 11:02:58</time></div><div class="recent-post-content">这节从OLS和SI转向MME
SI: 最小化预测Var(T-I)的误差方差OLS： 最小误差方差，最终由观测值（残差）的方差加权GLS： 最小化权重误差方差（最小二乘）， 参考观测值之间的协方差MME: mixed models, 同时最小化误差方差和random effectde 预测误差方差BLUP是由SI演变而来
Miexd model 是在康奈尔大学建立的， 主要由两个人建立： S.R. Searle and C.R. Henderson
一般的矩阵方程： y = Xβ + Zu + e             （1）y 为观测值vector（已知）； β 为fixed effects ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="遗传算法"> 遗传算法</a></div><div class="read-more"><a class="more" href="2020/05/23/MME/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/05/24/Learning_in_first_year/" target="_blank" rel="noopener">Learning in first year of PhD project</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-05-24T07:52:26.046Z">2020-05-24 09:52:26</time><i class="fa fa-edit"></i><time datetime="2020-05-24T07:52:26.046Z">2020-05-24 10:42:53</time></div><div class="recent-post-content">The first project(NUE AND NL)Clement suggested: predictions for NUE and NL were only used when the global H (GH), which is the standardized Mahalanobis distance between the prediction MIR record and the calibration data sets, were below or equal to 3.Nicolas suggested:        1 the selection of calv ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="经验"> 经验</a></div><div class="read-more"><a class="more" href="2020/05/24/Learning_in_first_year/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/04/08/LIUNX中的seq命令/" target="_blank" rel="noopener">生成指定位数的连续数字</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-04-08T14:45:58.754Z">2020-04-08 16:45:58</time><i class="fa fa-edit"></i><time datetime="2020-04-08T14:45:58.754Z">2020-04-08 16:48:09</time></div><div class="recent-post-content">有时候可能有这样的需要：用Shell生成类似0001这样的序列作为批次号，这里整理了一下个人的方法
方法一：通过seq命令seq命令可以生成从某个数字到递增到另一数字的序列。用法如下：
seq –helpUsage: seq [选项]　　尾数  or:  seq [选项]　　首数 尾数  or:  seq [选项]　　首数 增量值　尾数Print numbers from FIRST to LAST, in steps of INCREMENT.选项参数如下：
  -f, –format=格式      use printf style floating-point FORMAT  -s, – ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="shell"> shell</a></div><div class="read-more"><a class="more" href="2020/04/08/LIUNX中的seq命令/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/04/08/Liunx-split/" target="_blank" rel="noopener">Linux的split命令</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-04-08T07:14:15.428Z">2020-04-08 09:14:15</time><i class="fa fa-edit"></i><time datetime="2020-04-08T07:14:15.428Z">2020-04-08 09:22:36</time></div><div class="recent-post-content">1简介在Linux的coreutils中有一个工具split实现了这个功能，可以把一个大文件切割为多个小文件，甚至可以借助管道将流切成固定大小的文件，切割规则可以使用行数，也可以使用字节数。
2、预备测试数据先生成一个稍微大点的文件，接下来就切割这个文件：
#! /bin/bash
for((i=0;i&lt;1000000;i++));do    echo $i &gt;&gt; datadone将上面这段保存为gen-data.sh，然后执行生成一个名为data有1000000行的文本文件。
3、按行数分割将data按照每10000行为一个文件切割：split -l 10000 -d -a ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="Liunx"> Liunx</a></div><div class="read-more"><a class="more" href="2020/04/08/Liunx-split/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div></section><nav class="cxo-page-nav"><span class="page-number current">1</span><a class="page-number" href="page/2/" target="_blank" rel="noopener">2</a><a class="page-number" href="page/3/" target="_blank" rel="noopener">3</a><a class="extend next" rel="next noopener" href="page/2/" target="_blank">NEXT &amp;#62;</a></nav></section></article><footer id="cxo-footer-outer"><div id="cxo-footer-inner"><p class="footer-container"><span>Site by </span><a href="http://hexo.io" target="_blank" rel="noopener"><span>Hexo</span></a><span> | theme </span><a href="https://github.com/Longlongyu/hexo-theme-Cxo" target="_blank" rel="noopener"><span>Cxo</span></a></p><i class="fa fa-user"> </i><span id="busuanzi_value_site_uv"></span><span> | </span><i class="fa fa-eye"> </i><span id="busuanzi_value_site_pv"></span></div></footer><!-- catelog--><!-- top--><i class="fa fa-arrow-up close" id="go-up" aria-hidden="true"></i></body></html>