<!DOCTYPE html><html lang="zh-Hans"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no"><meta name="author" content="Yansen"><meta name="renderer" content="webkit"><meta name="copyright" content="Yansen"><meta name="keywords" content="Animal genetic"><meta name="description" content=""><meta name="Cache-Control" content="no-cache"><meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"><title>Animal Genetic</title><link rel="stylesheet" href="/css/style.css?v=2018.7.9"><link rel="stylesheet" href="/css/animation.css?v=2018.7.9"><link rel="icon" href="/img/assets/favicon.ico"><link rel="stylesheet" href="https://cdn.bootcss.com/font-awesome/4.7.0/css/font-awesome.min.css?version=1.5.6"><!-- scripts--><script>(function( w ){
  "use strict";
  // rel=preload support test
  if( !w.loadCSS ){
    w.loadCSS = function(){};
  }
  // define on the loadCSS obj
  var rp = loadCSS.relpreload = {};
  // rel=preload feature support test
  // runs once and returns a function for compat purposes
  rp.support = (function(){
    var ret;
    try {
      ret = w.document.createElement( "link" ).relList.supports( "preload" );
    } catch (e) {
      ret = false;
    }
    return function(){
      return ret;
    };
  })();

  // if preload isn't supported, get an asynchronous load by using a non-matching media attribute
  // then change that media back to its intended value on load
  rp.bindMediaToggle = function( link ){
    // remember existing media attr for ultimate state, or default to 'all'
    var finalMedia = link.media || "all";

    function enableStylesheet(){
      link.media = finalMedia;
    }

    // bind load handlers to enable media
    if( link.addEventListener ){
      link.addEventListener( "load", enableStylesheet );
    } else if( link.attachEvent ){
      link.attachEvent( "onload", enableStylesheet );
    }

    // Set rel and non-applicable media type to start an async request
    // note: timeout allows this to happen async to let rendering continue in IE
    setTimeout(function(){
      link.rel = "stylesheet";
      link.media = "only x";
    });
    // also enable media after 3 seconds,
    // which will catch very old browsers (android 2.x, old firefox) that don't support onload on link
    setTimeout( enableStylesheet, 3000 );
  };

  // loop through link elements in DOM
  rp.poly = function(){
    // double check this to prevent external calls from running
    if( rp.support() ){
      return;
    }
    var links = w.document.getElementsByTagName( "link" );
    for( var i = 0; i < links.length; i++ ){
      var link = links[ i ];
      // qualify links to those with rel=preload and as=style attrs
      if( link.rel === "preload" && link.getAttribute( "as" ) === "style" && !link.getAttribute( "data-loadcss" ) ){
        // prevent rerunning on link
        link.setAttribute( "data-loadcss", true );
        // bind listeners to toggle media back
        rp.bindMediaToggle( link );
      }
    }
  };

  // if unsupported, run the polyfill
  if( !rp.support() ){
    // run once at least
    rp.poly();

    // rerun poly on an interval until onload
    var run = w.setInterval( rp.poly, 500 );
    if( w.addEventListener ){
      w.addEventListener( "load", function(){
        rp.poly();
        w.clearInterval( run );
      } );
    } else if( w.attachEvent ){
      w.attachEvent( "onload", function(){
        rp.poly();
        w.clearInterval( run );
      } );
    }
  }


  // commonjs
  if( typeof exports !== "undefined" ){
    exports.loadCSS = loadCSS;
  }
  else {
    w.loadCSS = loadCSS;
  }
}( typeof global !== "undefined" ? global : this ) );</script><script src="https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js" defer></script><script src="/js/main.js?v=2018.7.9" defer></script><!-- fancybox--><link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'"><script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" defer></script><!-- busuanzi--><script async src="//dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script><meta name="generator" content="Hexo 4.2.0"></head><body><section class="profile-close" id="cxo-profile"><div class="profile-avatar"><i class="fa fa-caret-left"></i><img src="img/assets/cat.png"></div><!--.profile-saying
  i.fa.fa-comment
  .saying--><div class="cxo-profile-inner"><div class="profile-name">Animal Genetic</div><div class="profile-signature">LOVE</div><div class="friends"><div>FRIENDS</div><span><a href="//github.com/Yansen0515" target="_black">friendA</a></span><span><a href="//github.com/" target="_black">friendB</a></span><span><a href="//github.com/" target="_black">friendC</a></span></div><div class="read-progress"></div></div></section><header id="cxo-intro" style="height: 70vh;background-image: url(/img/intro/index-bg.png);"><nav id="cxo-intro-nav"><section><div class="intro-nav-title"><a href="/" target="_blank" rel="noopener">Yan&amp;Qing</a></div><div class="intro-nav-label-box"><a href="/" target="_blank" rel="noopener">Home</a><a href="/about/" target="_blank" rel="noopener">About</a><a href="/archives/" target="_blank" rel="noopener">Archives</a><a href="/tags/" target="_blank" rel="noopener">Tags</a><a href="/Categories/" target="_blank" rel="noopener">Categories</a></div><i class="fa fa-bars intro-nav-menu"><div class="intro-nav-drop"><a href="/" target="_blank" rel="noopener">Home</a><a href="/about/" target="_blank" rel="noopener">About</a><a href="/archives/" target="_blank" rel="noopener">Archives</a><a href="/tags/" target="_blank" rel="noopener">Tags</a><a href="/Categories/" target="_blank" rel="noopener">Categories</a></div></i><div class="clear"></div></section></nav><h1>Yan&amp;Qing</h1><h5>Welcome to my Blog!</h5><div class="intro-social"><a class="fa fa-envelope" href="mailto:yansen0515@gmail" title="email"></a><a class="fa fa-github" href="//github.com/Yansen0515" target="_blank" title="github"></a><a class="fa fa-weibo" href="//www.weibo.com" target="_blank" title="weibo"></a><a class="fa fa-facebook" href="//www.facebook.com/Yansen Chen" target="_blank" title="facebook"></a><a class="fa fa-twitter" href="//twitter.com" target="_blank" title="twitter"></a><a class="fa fa-rss" href="/atom.xml" target="_blank" title="rss"></a></div></header><article class="cxo-up" id="cxo-content-outer"><section id="cxo-content-inner"><section id="recent-posts"><div class="recent-post-item"><div class="post-title"><a href="2020/02/26/GLM/" target="_blank" rel="noopener">GLM</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-02-26T10:31:00.281Z">2020-02-26 11:31:00</time><i class="fa fa-edit"></i><time datetime="2020-02-26T10:31:00.281Z">2020-05-26 16:43:42</time></div><div class="recent-post-content">前言之前我们使用R进行线性分析时，我们得到了满足正态分布下的线性拟合模型，但在许多实际情况中，假定满足正态分布的假设并不合理结果变量为类别型：包括二值变量（是/否、通过/失败、存活/死亡）和多类别变量（优/良/可/差）等。结果变量为计数型的（一周交通事故的数目、每日酒水消耗的数量），这类变量都是非负的有限值，它们的均值和方差通常都是相关的（而正态分布变量间则是相互独立）。为此，我们在一般线性的基础上进行深入学习，来研究包含了非正态因变量的广义线性模型。
一、GLM函数概况在R语言中，我们使用glm()函数构建广义线性模型，调用语法如下：
family为拟合所属的函数族function为对应的连 ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="算法"> 算法</a></div><div class="read-more"><a class="more" href="2020/02/26/GLM/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/05/20/OLSandSI/" target="_blank" rel="noopener">1 - OLS and SI</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-05-20T20:28:23.874Z">2020-05-20 22:28:23</time><i class="fa fa-edit"></i><time datetime="2020-05-20T20:28:23.874Z">2020-05-25 20:46:55</time></div><div class="recent-post-content">前言： 对自己上半年的一个遗传课程“Genetics, Genomics and (Bio)diversity”整理，其是Liege university 的Nicolas Gengler主讲课程都是由Octave语言，完成练习。多使用矩阵计算，需要相关基础知识：如：Kronecker product; Cholsky分解； Cannoical分解； 广义逆，
###############first part OLS简单的线性回归模型：y = b1 + b2*xb1为截距，b2为斜率
矩阵写法： y = Wb + e    （1）
求解时： 我们需要使e最小此方法被称为： Least squ ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="遗传算法"> 遗传算法</a></div><div class="read-more"><a class="more" href="2020/05/20/OLSandSI/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/04/08/计算大约PEV/" target="_blank" rel="noopener">计算大约PEV ：</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-04-08T05:29:46.441Z">2020-04-08 07:29:46</time><i class="fa fa-edit"></i><time datetime="2020-04-08T05:29:46.441Z">2020-04-13 09:55:43</time></div><div class="recent-post-content">数据太大，BLUPF90无法直接求出SE因修改的软件，每次只能进行一个动物的PEV计算，找出具有大于100女儿数的公牛查看系谱的父亲的女儿数：awk ‘{print $2}’ ped.txt | sort +0 -1 | uniq -c &gt; sire.prog
###与renum数据整合，得到level.均要先sort, 再join:join -1 +1 -2 +2 sir.tem2 pediall.tem | sort +1 -2 &gt; sir200pedi.tem
####也可以直接在renum系谱中，查看女儿数（推荐）
awk ‘$8&gt;100{print $1,$10,$ ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="shell"> shell</a></div><div class="read-more"><a class="more" href="2020/04/08/计算大约PEV/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/04/08/LIUNX中的seq命令/" target="_blank" rel="noopener">生成指定位数的连续数字</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-04-08T14:45:58.754Z">2020-04-08 16:45:58</time><i class="fa fa-edit"></i><time datetime="2020-04-08T14:45:58.754Z">2020-04-08 16:48:09</time></div><div class="recent-post-content">有时候可能有这样的需要：用Shell生成类似0001这样的序列作为批次号，这里整理了一下个人的方法
方法一：通过seq命令seq命令可以生成从某个数字到递增到另一数字的序列。用法如下：
seq –helpUsage: seq [选项]　　尾数  or:  seq [选项]　　首数 尾数  or:  seq [选项]　　首数 增量值　尾数Print numbers from FIRST to LAST, in steps of INCREMENT.选项参数如下：
  -f, –format=格式      use printf style floating-point FORMAT  -s, – ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="shell"> shell</a></div><div class="read-more"><a class="more" href="2020/04/08/LIUNX中的seq命令/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/04/08/Liunx-split/" target="_blank" rel="noopener">Linux的split命令</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-04-08T07:14:15.428Z">2020-04-08 09:14:15</time><i class="fa fa-edit"></i><time datetime="2020-04-08T07:14:15.428Z">2020-04-08 09:22:36</time></div><div class="recent-post-content">1简介在Linux的coreutils中有一个工具split实现了这个功能，可以把一个大文件切割为多个小文件，甚至可以借助管道将流切成固定大小的文件，切割规则可以使用行数，也可以使用字节数。
2、预备测试数据先生成一个稍微大点的文件，接下来就切割这个文件：
#! /bin/bash
for((i=0;i&lt;1000000;i++));do    echo $i &gt;&gt; datadone将上面这段保存为gen-data.sh，然后执行生成一个名为data有1000000行的文本文件。
3、按行数分割将data按照每10000行为一个文件切割：split -l 10000 -d -a ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="Liunx"> Liunx</a></div><div class="read-more"><a class="more" href="2020/04/08/Liunx-split/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/04/01/计算重复力模型EBV与其他性状遗传相关的R代码/" target="_blank" rel="noopener">计算重复力模型EBV与其他性状遗传相关的R代码</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-04-01T17:25:04.417Z">2020-04-01 19:25:04</time><i class="fa fa-edit"></i><time datetime="2020-04-01T17:25:04.417Z">2020-04-01 19:26:19</time></div><div class="recent-post-content">#####2019-12-2 22:47:45#####correlation by EBV######setwd(“C:\Users\Desktop\PhDdata\result\EBV”)library(data.table)EBV &lt;- fread(“newsolutions”,header=T)head(EBV)acEBV &lt;- EBV[effect==8]
trait1 &lt;- acEBV[trait==1]trait2 &lt;- acEBV[trait==2]trait3 &lt;- acEBV[trait==3]trait4 &lt;- acEBV[trait= ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="R"> R</a></div><div class="read-more"><a class="more" href="2020/04/01/计算重复力模型EBV与其他性状遗传相关的R代码/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/04/01/计算重复力模型结果的遗传力及相关R代码/" target="_blank" rel="noopener">计算重复力模型结果的遗传力及相关R代码</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-04-01T08:29:35.424Z">2020-04-01 10:29:35</time><i class="fa fa-edit"></i><time datetime="2020-04-01T08:29:35.424Z">2020-04-01 10:30:56</time></div><div class="recent-post-content">param= function(){
  GA=as.matrix(read.table(“GA.csv”,h=F,sep=”,”))  #PE  PE=as.matrix(read.table(“PE.csv”,h=F,sep=”,”))  #Prity3PE  Parity3PE =as.matrix(read.table(“Parity3PE.csv”,h=F,sep=”,”))  #CP  #CP=as.matrix(read.table(“CP.csv”,h=F,sep=”,”))  #Residual  R=as.matrix(read.table(“R.csv”,h=F,sep= ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="R"> R</a></div><div class="read-more"><a class="more" href="2020/04/01/计算重复力模型结果的遗传力及相关R代码/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/03/27/octave计算遗传力的画图代码/" target="_blank" rel="noopener">Octave计算遗传力画图R代码</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-03-27T08:12:15.669Z">2020-03-27 09:12:15</time><i class="fa fa-edit"></i><time datetime="2020-03-27T08:12:15.669Z">2020-03-27 09:15:06</time></div><div class="recent-post-content">#############MUsetwd(“C:\Users\chenys\Desktop\PhDdata\MU\jieguo”)data &lt;- read.table(“MUresults3RF.csv”,sep=”,”,header=T,fill=TRUE)tail(data)library(Hmisc)library(ggplot2)##herd x year of calvingpar(font=2,las=1,mar = c(5,5,4,2) + 0.1)plot(data$DIM, data$SDYY.1, xlab =””, ylab = “ “ ,type=”o”,lwd= ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="R"> R</a></div><div class="read-more"><a class="more" href="2020/03/27/octave计算遗传力的画图代码/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/03/10/liunx-2/" target="_blank" rel="noopener">Liunx(2)&amp;BLUPF90</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-03-10T20:41:34.202Z">2020-03-10 21:41:34</time><i class="fa fa-edit"></i><time datetime="2020-03-10T20:41:34.202Z">2020-03-10 21:42:55</time></div><div class="recent-post-content">查看大的文件：less –S filewc -l genotype.txt pedigree.txt
expand             replaces TAB with spacesuniq                    retains unique lines on a sorted file


代表为在file后面添加数据Cat 是以列相叠加cat file1 file2 &gt; output_filePase 以行进行连接，但是以TAB连接，需要使用expand处理，其会使tab变成空格Eg: pase fileA fileB | expand | combind C但 ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="Liunx&amp;BLUPF90"> Liunx&amp;BLUPF90</a></div><div class="read-more"><a class="more" href="2020/03/10/liunx-2/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div><div class="recent-post-item"><div class="post-title"><a href="2020/02/26/annovar使用/" target="_blank" rel="noopener">annovar注释软件</a></div><div class="post-time"><i class="fa fa-table"></i><time datetime="2020-02-26T21:32:56.789Z">2020-02-26 22:32:56</time><i class="fa fa-edit"></i><time datetime="2020-02-26T21:32:56.789Z">2020-02-26 22:48:27</time></div><div class="recent-post-content">##annovar解压包ANNOVAR│  annotate_variation.pl #主程序，功能包括下载数据库，三种不同的注释│  coding_change.pl #可用来推断蛋白质序列│  convert2annovar.pl #将多种格式转为.avinput的程序│  retrieve_seq_from_fasta.pl #用于自行建立其他物种的转录本│  table_annovar.pl #注释程序，可一次性完成三种类型的注释│  variants_reduction.pl #可用来更灵活地定制过滤注释流程│├─example #存放示例文件│└─humandb #人类注释数据库 ...</div><div class="recent-post-tags"><a class="post-tag fa fa-tag" href="javascript:void(0)" date-tags="perl"> perl</a></div><div class="read-more"><a class="more" href="2020/02/26/annovar使用/#more" target="_blank" rel="noopener">ReadMore</a></div><hr></div></section><nav class="cxo-page-nav"><a class="extend prev" rel="prev noopener" href="">&amp;#60; PREV</a><a class=" target="_blank"page-number" href="">1</a><span class="page-number current">2</span><a class="page-number" href="page/3/" target="_blank" rel="noopener">3</a><a class="extend next" rel="next noopener" href="page/3/" target="_blank">NEXT &amp;#62;</a></nav></section></article><footer id="cxo-footer-outer"><div id="cxo-footer-inner"><p class="footer-container"><span>Site by </span><a href="http://hexo.io" target="_blank" rel="noopener"><span>Hexo</span></a><span> | theme </span><a href="https://github.com/Longlongyu/hexo-theme-Cxo" target="_blank" rel="noopener"><span>Cxo</span></a></p><i class="fa fa-user"> </i><span id="busuanzi_value_site_uv"></span><span> | </span><i class="fa fa-eye"> </i><span id="busuanzi_value_site_pv"></span></div></footer><!-- catelog--><!-- top--><i class="fa fa-arrow-up close" id="go-up" aria-hidden="true"></i></body></html>