<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 4.2.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="css/main.css">


<link rel="stylesheet" href="lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"Yansen0515.github.io.git","root":"/","scheme":"Mist","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>

  <meta property="og:type" content="website">
<meta property="og:title" content="Animal genetic">
<meta property="og:url" content="page/2/index.html">
<meta property="og:site_name" content="Animal genetic">
<meta property="article:author" content="Yansen">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="page/2/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : true,
    isPost : false,
    lang   : 'zh-Hans'
  };
</script>

  <title>Animal genetic</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="تشغيل شريط التصفح">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" target="_blank" class="brand" rel="start noopener">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Animal genetic</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">Follow your heart</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" target="_blank" rel="section noopener"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" target="_blank" rel="section noopener"><i class="fa fa-archive fa-fw"></i>الأرشيفات</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content index posts-expand">
            
      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="2020/04/08/计算大约PEV/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="images/avatar.gif">
      <meta itemprop="name" content="Yansen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Animal genetic">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="2020/04/08/%E8%AE%A1%E7%AE%97%E5%A4%A7%E7%BA%A6PEV/" target="_blank" rel="noopener" class="post-title-link" itemprop="url">计算大约PEV ：</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">نُشر في</span>

              <time title="أُنشأ: 2020-04-08 07:29:46" itemprop="dateCreated datePublished" datetime="2020-04-08T07:29:46+02:00">2020-04-08</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">عُدل في</span>
                <time title="عُدل: 2020-04-13 09:55:43" itemprop="dateModified" datetime="2020-04-13T09:55:43+02:00">2020-04-13</time>
              </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>数据太大，BLUPF90无法直接求出SE<br>因修改的软件，每次只能进行一个动物的PEV计算，找出具有大于100女儿数的公牛<br>查看系谱的父亲的女儿数：<br>awk ‘{print $2}’ ped.txt | sort +0 -1 | uniq -c &gt; sire.prog</p>
<p>###与renum数据整合，得到level.<br>均要先sort, 再join:<br>join -1 +1 -2 +2 sir.tem2 pediall.tem | sort +1 -2 &gt; sir200pedi.tem</p>
<p>####也可以直接在renum系谱中，查看女儿数（推荐）</p>
<p>awk ‘$8&gt;100{print $1,$10,$8}’ renadd02.ped &gt; sir100.tem<br>My supervisor give me advice for interbull:<br>“As 10 daughters with at least 3 TD records in 10 different herds (I think this is a minimum standard we use for INTERBULL) minimum standards on the side of INTERBULL traits etc…. all this to limit the number of runs.  ”</p>
<p>###得到interbull后与本组数据库匹配，得到共同interbull  !!!!!!!!!!</p>
<p>#################<br>查看女儿数后，查看女儿在女场的分布情况（）<br>计算大约PEV ：<br>数据切割： split -l 9 rhs.all -d -a 4 rhs_<br>往文件中，加入标题：ls rhs_* |while read id; do sed ‘1i\trait\effect level rhs’    $id &gt; new$id ;done</p>
<p>生成相应的parametre file : ls newrhs* |while read id; do sed “s/rhs.dat/$id/“ renblup.par &gt; $id.renblup.par ;done</p>
<p>进行bash循环：</p>
<p>ls  *renblup.par |while read id; do echo $id &gt;run &amp;&amp; /home/ULG/f031707/progs/blup90iod2 &lt; run &gt; blupiod.$id.out &amp;&amp; mv solutions $id.solution ;done</p>
<p>生成PEV：<br>ls *solution | while read id ; do awk ‘NR==FNR{if(NR&gt;1) a[$1, $2, $3] = $4; next} {if (NR &gt;1) print $1,$2,$3,$4-a[$1, $2, $3] }’ solutionsold $id &gt; $id.pev；done</p>
<p>提出需要的PEV：<br>##############直接全部<br>for i in $(seq –w 0 227); do id=$(sed -n “2p”  newrhs_0${i} |awk ‘{print $3}’) &amp;&amp; grep -w  “$id” newrhs_0${i}.renblup.par.solution.pev &gt;newrhs_0${i}.renblup.par.solution.pev.extract ;done</p>
<p>##############分开<br>##1-9<br>for i in $(seq 1 9); do id=$(sed -n “2p”  newrhs_000${i} |awk ‘{print $3}’) &amp;&amp; grep -w  “$id” newrhs_000${i}.renblup.par.solution.pev &gt;newrhs_000${i}.renblup.par.solution.pev.extract ;done</p>
<p>#10-99<br>for i in $(seq 10 99); do id=$(sed -n “2p”  newrhs_00${i} |awk ‘{print $3}’) &amp;&amp; grep -w  “$id” newrhs_00${i}.renblup.par.solution.pev &gt;newrhs_00${i}.renblup.par.solution.pev.extract ;done</p>
<p>#100-228</p>
<p>for i in $(seq 100 227); do id=$(sed -n “2p”  newrhs_0${i} |awk ‘{print $3}’) &amp;&amp; grep -w  “$id” newrhs_0${i}.renblup.par.solution.pev &gt;newrhs_0${i}.renblup.par.solution.pev.extract ;done</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="2020/04/01/计算重复力模型EBV与其他性状遗传相关的R代码/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="images/avatar.gif">
      <meta itemprop="name" content="Yansen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Animal genetic">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="2020/04/01/%E8%AE%A1%E7%AE%97%E9%87%8D%E5%A4%8D%E5%8A%9B%E6%A8%A1%E5%9E%8BEBV%E4%B8%8E%E5%85%B6%E4%BB%96%E6%80%A7%E7%8A%B6%E9%81%97%E4%BC%A0%E7%9B%B8%E5%85%B3%E7%9A%84R%E4%BB%A3%E7%A0%81/" target="_blank" rel="noopener" class="post-title-link" itemprop="url">计算重复力模型EBV与其他性状遗传相关的R代码</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">نُشر في</span>
              

              <time title="أُنشأ: 2020-04-01 19:25:04 / عُدل: 19:26:19" itemprop="dateCreated datePublished" datetime="2020-04-01T19:25:04+02:00">2020-04-01</time>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>#####2019-12-2 22:47:45<br>#####correlation by EBV<br>#####<br>#<br>setwd(“C:\Users\Desktop\PhDdata\result\EBV”)<br>library(data.table)<br>EBV &lt;- fread(“newsolutions”,header=T)<br>head(EBV)<br>acEBV &lt;- EBV[effect==8]</p>
<p>trait1 &lt;- acEBV[trait==1]<br>trait2 &lt;- acEBV[trait==2]<br>trait3 &lt;- acEBV[trait==3]<br>trait4 &lt;- acEBV[trait==4]<br>head(trait2)<br>#4个性状的EBV and SE<br>use_EBV &lt;- cbind(trait1, trait2[,c(4,5)], trait3[,c(4,5)], trait4[,c(4,5)])<br>names(use_EBV)[4:11] &lt;- c(“sol1”,”SE1”,”sol2”,”SE2”,”sol3”,”SE3”,”sol4”,”SE4”)<br>head(use_EBV)</p>
<p>#计算repeatabilty///reliability    1- (SE^2 / va^2)<br>use_EBV$rel1 &lt;- 1-((use_EBV[,5])^2)/3.48<br>use_EBV$rel2 &lt;- 1-((use_EBV[,7])^2)/2.10E-04<br>use_EBV$rel3 &lt;- 1-((use_EBV[,9])^2)/4.755<br>use_EBV$rel4 &lt;- 1-((use_EBV[,11])^2)/3.40E-04</p>
<p>###读入renum系谱<br>ped &lt;- fread(“newrenadd05.ped”)<br>comped &lt;- ped[,c(1,10)]<br>names(comped) &lt;- c(“level”,”ID”)<br>head(comped)<br>###加入ID<br>add_ID &lt;- merge(use_EBV, comped, by.x=”level”, by.y = “level”)<br>###修改ID，与InterBull相同格式<br>add_ID$ID_c &lt;- substr(add_ID$ID,4,19)<br>head(add_ID)<br>write.csv(add_ID,”Sol_addID.csv”,row.names = F)</p>
<p>interbull &lt;- add_ID[,c(17,4,6,8,10,12:15)]<br>head(interbull)<br>####自己性状reliablity大于0.5<br>use_interbull &lt;- interbull[interbull$rel1&gt;0.5 &amp; interbull$rel2&gt;0.5,]<br>use_interbull2 &lt;- use_interbull[use_interbull$rel3&gt;0.5 &amp;use_interbull$rel4&gt;0.5, ]<br>head(use_interbull2)<br>write.table(use_interbull2, file =”use_interbull.csv”,row.names = F,quote=FALSE)</p>
<p>#####计算相关性</p>
<p>EBV_all1 &lt;- fread(“all_product.csv”, header = T)<br>head(EBV_all1[,1:10])<br>EBV_all2 &lt;- EBV_all1[,c(1,5,9,10:27)]<br>head(EBV_all2[1:3])<br>##画所有EBV的直接相关图<br>EBV1 &lt;- EBV_all2[,c(2,5:9,11,13,15,17,19,21)]</p>
<p>r &lt;- cor(EBV1)<br>library(corrplot)<br>par(mfrow = c(1,1))<br>corrplot(r,method=’pie’,type = “lower”)</p>
<p>rel_1 &lt;-  EBV_all2[,c(4,10,12,14,16,18,20)]/100<br>rel_1 &lt;- as.data.frame(rel_1)<br>rel_all &lt;- cbind(EBV_all2$reb4, rel_1)</p>
<p>head(rel_all)<br>NUE1_Milk &lt;- sqrt(sum(EBV_all2$reb4)<em>sum(rel_1[,1]))</em>r[1,2]/sum(EBV_all2$reb4 * rel_1[,1])<br>NUE1_Milk<br>cor_Production = c()<br>for (i in 2:6){<br>  cor_Pro = sqrt(sum(EBV_all2[,3])<em>sum(rel_1$Pro_rel))</em>r[1,i]/sum(EBV_all2[,3] * rel_1$Pro_rel)<br>  cor_Production = cbind(cor_Production, cor_Pro)<br>  }<br>cor_Production<br>##NUE1_Milk_2 &lt;- sqrt(mean(EBV_all1$reb1)<em>mean(rel_1$Pro_rel)) *r[1,2] /mean(EBV_all1$reb1</em>rel_1$Pro_rel)<br>NUE1_SCS &lt;- sqrt(sum(EBV_all2$reb4)<em>sum(rel_1[,2]))</em>r[1,7]/sum(EBV_all2$reb4 * rel_1[,2])<br>NUE1_SCS<br>dim(rel_1)<br>cor_list = c()<br>for (k in 2:7) {<br>    cor_list2 = sqrt(sum(EBV_all2[,3])*sum(rel_1[,k])) * r[1,k+5]/sum(EBV_all2[,3] * rel_1[,k])</p>
<pre><code># cor_list2 = sqrt(sum(EBV_all1[,3])*sum(rel_1[,parse(tmp)]))*rr/sum(EBV_all1[,3] * rel_1[,parse(tmp)])</code></pre><p>  cor_list = cbind(cor_list,cor_list2)<br>}<br>cor_list</p>
<p>  ########分解 解答</p>
<h1 id=""><a href="#" target="_blank" rel="noopener" class="headerlink" title=""></a></h1><h1 id="NUE1-long-sqrt-sum-EBV-all1-3-sum-rel-1-3-r-1-8-sum-EBV-all1-3-rel-1-3"><a href="#NUE1-long-sqrt-sum-EBV-all1-3-sum-rel-1-3-r-1-8-sum-EBV-all1-3-rel-1-3" target="_blank" rel="noopener" class="headerlink" title="NUE1_long = sqrt(sum(EBV_all1[,3])sum(rel_1[,3]))r[1,8]/sum(EBV_all1[,3] * rel_1[,3])"></a>NUE1_long = sqrt(sum(EBV_all1[,3])<em>sum(rel_1[,3]))</em>r[1,8]/sum(EBV_all1[,3] * rel_1[,3])</h1><h1 id="NUE1-ferti-sqrt-sum-EBV-all1-3-sum-rel-1-4-r-1-9-sum-EBV-all1-3-rel-1-4"><a href="#NUE1-ferti-sqrt-sum-EBV-all1-3-sum-rel-1-4-r-1-9-sum-EBV-all1-3-rel-1-4" target="_blank" rel="noopener" class="headerlink" title="NUE1_ferti = sqrt(sum(EBV_all1[,3])sum(rel_1[,4]))r[1,9]/sum(EBV_all1[,3] * rel_1[,4])"></a>NUE1_ferti = sqrt(sum(EBV_all1[,3])<em>sum(rel_1[,4]))</em>r[1,9]/sum(EBV_all1[,3] * rel_1[,4])</h1><h1 id="NUE1-BCS-sqrt-sum-EBV-all1-3-sum-rel-1-5-r-1-10-sum-EBV-all1-3-rel-1-5"><a href="#NUE1-BCS-sqrt-sum-EBV-all1-3-sum-rel-1-5-r-1-10-sum-EBV-all1-3-rel-1-5" target="_blank" rel="noopener" class="headerlink" title="NUE1_BCS = sqrt(sum(EBV_all1[,3])sum(rel_1[,5]))r[1,10]/sum(EBV_all1[,3] * rel_1[,5])"></a>NUE1_BCS = sqrt(sum(EBV_all1[,3])<em>sum(rel_1[,5]))</em>r[1,10]/sum(EBV_all1[,3] * rel_1[,5])</h1><h1 id="NUE1-DCE-sqrt-sum-EBV-all1-3-sum-rel-1-6-r-1-11-sum-EBV-all1-3-rel-1-6"><a href="#NUE1-DCE-sqrt-sum-EBV-all1-3-sum-rel-1-6-r-1-11-sum-EBV-all1-3-rel-1-6" target="_blank" rel="noopener" class="headerlink" title="NUE1_DCE = sqrt(sum(EBV_all1[,3])sum(rel_1[,6]))r[1,11]/sum(EBV_all1[,3] * rel_1[,6])"></a>NUE1_DCE = sqrt(sum(EBV_all1[,3])<em>sum(rel_1[,6]))</em>r[1,11]/sum(EBV_all1[,3] * rel_1[,6])</h1><h1 id="NUE1-MCE-sqrt-sum-EBV-all1-3-sum-rel-1-7-r-1-12-sum-EBV-all1-3-rel-1-7"><a href="#NUE1-MCE-sqrt-sum-EBV-all1-3-sum-rel-1-7-r-1-12-sum-EBV-all1-3-rel-1-7" target="_blank" rel="noopener" class="headerlink" title="NUE1_MCE = sqrt(sum(EBV_all1[,3])sum(rel_1[,7]))r[1,12]/sum(EBV_all1[,3] * rel_1[,7])"></a>NUE1_MCE = sqrt(sum(EBV_all1[,3])<em>sum(rel_1[,7]))</em>r[1,12]/sum(EBV_all1[,3] * rel_1[,7])</h1><h1 id="NUE1-SCS"><a href="#NUE1-SCS" target="_blank" rel="noopener" class="headerlink" title="NUE1_SCS"></a>NUE1_SCS</h1><h1 id="NUE1-long"><a href="#NUE1-long" target="_blank" rel="noopener" class="headerlink" title="NUE1_long"></a>NUE1_long</h1><h1 id="NUE1-ferti"><a href="#NUE1-ferti" target="_blank" rel="noopener" class="headerlink" title="NUE1_ferti"></a>NUE1_ferti</h1><h1 id="NUE1-BCS"><a href="#NUE1-BCS" target="_blank" rel="noopener" class="headerlink" title="NUE1_BCS"></a>NUE1_BCS</h1><h1 id="NUE1-DCE"><a href="#NUE1-DCE" target="_blank" rel="noopener" class="headerlink" title="NUE1_DCE"></a>NUE1_DCE</h1><h1 id="NUE1-MCE"><a href="#NUE1-MCE" target="_blank" rel="noopener" class="headerlink" title="NUE1_MCE"></a>NUE1_MCE</h1><p>###############cor type </p>
<p>EBV_type &lt;- fread(“all_type.csv”)<br>head(EBV_type[,1:10])<br>EBV_type2 &lt;- EBV_type[,c(1,5,9,10:75)]<br>head(EBV_type2[,1:10])<br>EBV2 &lt;- EBV_type2[,c(2, 5,  7,  9, 11, 13, 15, 17, 19, 21 ,23 ,25, 27, 29, 31,<br>                    33, 35, 37, 39, 41, 43, 45, 47, 49, 51, 53, 55,57,59, 61,<br>                    63, 65, 67,69)]<br>r_type &lt;- cor(EBV2)<br>dim(EBV_type2)<br>library(corrplot)<br>par(mfrow = c(1,1))<br>corrplot(r_type,method=’pie’,type = “lower”)</p>
<p>sele_rel &lt;- seq(4,68,2)<br>sele_rel<br>rel_type &lt;-  EBV_type2[,c( 4,  6,  8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30,<br>                          32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54,<br>                          56, 58, 60, 62, 64, 66, 68)]/100<br>rel_type &lt;- as.data.frame(rel_type)<br>rel_all2 &lt;- cbind(EBV_type2$reb4, rel_type)<br>head(rel_all2)<br>NUE1_stature &lt;- sqrt(sum(rel_all2[,1])<em>sum(rel_all2[,2]))</em>r_type[1,2]/sum(rel_all2[,1] * rel_all2[,2])<br>NUE1_stature<br>dim(r_type)<br>##use  P.J. Blanchard ways<br>cor_type = c()<br>for (k in 2:34) {</p>
<p>  cor_lis2 = sqrt(sum(rel_all2[,1])*sum(rel_all2[,k])) * r_type[1,k]/sum(rel_all2[,1] * rel_all2[,k])</p>
<p>  cor_type = cbind(cor_type, cor_lis2)<br>}<br>cor_type</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="2020/04/01/计算重复力模型结果的遗传力及相关R代码/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="images/avatar.gif">
      <meta itemprop="name" content="Yansen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Animal genetic">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="2020/04/01/%E8%AE%A1%E7%AE%97%E9%87%8D%E5%A4%8D%E5%8A%9B%E6%A8%A1%E5%9E%8B%E7%BB%93%E6%9E%9C%E7%9A%84%E9%81%97%E4%BC%A0%E5%8A%9B%E5%8F%8A%E7%9B%B8%E5%85%B3R%E4%BB%A3%E7%A0%81/" target="_blank" rel="noopener" class="post-title-link" itemprop="url">计算重复力模型结果的遗传力及相关R代码</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">نُشر في</span>
              

              <time title="أُنشأ: 2020-04-01 10:29:35 / عُدل: 10:30:56" itemprop="dateCreated datePublished" datetime="2020-04-01T10:29:35+02:00">2020-04-01</time>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>param= function(){</p>
<p>  GA=as.matrix(read.table(“GA.csv”,h=F,sep=”,”))<br>  #PE<br>  PE=as.matrix(read.table(“PE.csv”,h=F,sep=”,”))<br>  #Prity3PE<br>  Parity3PE =as.matrix(read.table(“Parity3PE.csv”,h=F,sep=”,”))<br>  #CP<br>  #CP=as.matrix(read.table(“CP.csv”,h=F,sep=”,”))<br>  #Residual<br>  R=as.matrix(read.table(“R.csv”,h=F,sep=”,”))</p>
<p>  #Phenotypic Variance<br>  P1 = as.matrix(GA+PE+Parity3PE+R)</p>
<p>  #Heritabilities<br>  h2 = as.matrix((diag(GA)/diag(P1)))<br>  #PE ratio</p>
<p>  r =as.matrix(diag(GA+PE+Parity3PE) / diag(P1))<br>  ##Genetic_correlation GC<br>  diagGC=diag(x = 1,  nrow(GA),  ncol(GA), names = TRUE)<br>  diag(diagGC) &lt;- sqrt(diag(GA))<br>  GC=solve(diagGC) %<em>% GA %</em>% solve(t(diagGC))<br>  #PE_correlation PEC<br>  diagPEC=diag(x = 1,  nrow(PE),  ncol(PE), names = TRUE)<br>  diag(diagPEC) &lt;- sqrt(diag(PE+Parity3PE))<br>  PEC = solve(diagPEC)%<em>% (PE+Parity3PE) %</em>% solve(t(diagPEC))<br>  #Phenotype_correlation PC<br>  diagPC=diag(x = 1,  nrow(P1),  ncol(P1), names = TRUE)<br>  diag(diagPC) &lt;- sqrt(diag(P1))<br>  PC = solve(diagPC)%<em>% P1 %</em>% solve(t(diagPC))</p>
<p>return(list(h2,r,GC,PC))<br>}</p>
<p>param()</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="2020/03/27/octave计算遗传力的画图代码/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="images/avatar.gif">
      <meta itemprop="name" content="Yansen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Animal genetic">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="2020/03/27/octave%E8%AE%A1%E7%AE%97%E9%81%97%E4%BC%A0%E5%8A%9B%E7%9A%84%E7%94%BB%E5%9B%BE%E4%BB%A3%E7%A0%81/" target="_blank" rel="noopener" class="post-title-link" itemprop="url">Octave计算遗传力画图R代码</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">نُشر في</span>
              

              <time title="أُنشأ: 2020-03-27 09:12:15 / عُدل: 09:15:06" itemprop="dateCreated datePublished" datetime="2020-03-27T09:12:15+01:00">2020-03-27</time>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>#############MU<br>setwd(“C:\Users\chenys\Desktop\PhDdata\MU\jieguo”)<br>data &lt;- read.table(“MUresults3RF.csv”,sep=”,”,header=T,fill=TRUE)<br>tail(data)<br>library(Hmisc)<br>library(ggplot2)<br>##herd x year of calving<br>par(font=2,las=1,mar = c(5,5,4,2) + 0.1)<br>plot(data$DIM, data$SDYY.1, xlab =””, ylab = “ “ ,type=”o”,lwd=0.8,<br>     col=”blue”,lty=1, xlim = c(5,365), ylim=c(min(data$SDYY.1),max(data$SDYY.2)))<br>par(new=TRUE)<br>plot(data$DIM, data$SDYY.2, xlab =” “, ylab = “ “ ,type=”o”,lwd=0.8,<br>     col=”forestgreen”,lty=1, xlim = c(5,365), ylim=c(min(data$SDYY.1),max(data$SDYY.2)))<br>par(new=TRUE)<br>plot(data$DIM, data$SDYY.3, xlab =” “, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”red”,lty=1, xlim = c(5,365), ylim=c(min(data$SDYY.1),max(data$SDYY.2)))</p>
<p>minor.tick(nx=2, ny=2,tick.ratio=0.75)<br>par(mar = c(5,4,4,2) + 0.1)<br>title( xlab=”Days in Milk”,ylab=”HY variance”,cex.lab=1,font.lab=2,main = “Milk Urea” )<br>legend(0,20,bty=”0”, legend=c(“Parity1”,”Parity2”, “Parity3”),text.col=c(“blue”,”forestgreen”,”red”),<br>       col=c(“blue”,”forestgreen”,”red”),lty=c(1),cex=1.1,lwd=0.8, pch=c(1))</p>
<p>##PE<br>par(font=2,las=1,mar = c(5,5,4,2) + 0.1)<br>plot(data$DIM, data$SDWW.1, xlab =””, ylab = “ “ ,type=”o”,lwd=0.8,<br>     col=”blue”,lty=1, xlim = c(5,365), ylim=c(min(data$SDWW.1),max(data$SDWW.1)))<br>par(new=TRUE)<br>plot(data$DIM, data$SDWW.2, xlab =” “, ylab = “ “ ,type=”o”,lwd=0.8,<br>     col=”forestgreen”,lty=1, xlim = c(5,365), ylim=c(min(data$SDWW.1),max(data$SDWW.1)))<br>par(new=TRUE)<br>plot(data$DIM, data$SDWW.3, xlab =” “, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”red”,lty=1, xlim = c(5,365), ylim=c(min(data$SDWW.1),max(data$SDWW.1)))</p>
<p>minor.tick(nx=2, ny=2,tick.ratio=0.75)<br>par(mar = c(5,4,4,2) + 0.1)<br>title( xlab=”Days in Milk”,ylab=”PE variance”,cex.lab=1,font.lab=2,main = “Milk Urea” )<br>legend(0,16.5,bty=”0”, legend=c(“Parity1”,”Parity2”, “Parity3”),text.col=c(“blue”,”forestgreen”,”red”),<br>       col=c(“blue”,”forestgreen”,”red”),lty=c(1),cex=1.1,lwd=0.8, pch=c(1))</p>
<p>##AG<br>par(font=2,las=1,mar = c(5,5,4,2) + 0.1)<br>plot(data$DIM, data$SDGG.1, xlab =””, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”blue”,lty=1, xlim = c(5,365), ylim=c(min(data$SDGG.1),max(data$SDGG.1)))<br>par(new=TRUE)<br>plot(data$DIM, data$SDGG.2, xlab =” “, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”forestgreen”,lty=1, xlim = c(5,365), ylim=c(min(data$SDGG.1),max(data$SDGG.1)))<br>par(new=TRUE)<br>plot(data$DIM, data$SDGG.3, xlab =” “, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”red”,lty=1, xlim = c(5,365), ylim=c(min(data$SDGG.1),max(data$SDGG.1)))</p>
<p>minor.tick(nx=2, ny=2,tick.ratio=0.75)<br>par(mar = c(5,4,4,2) + 0.1)<br>title(xlab=”Days in Milk”,ylab=”Genetic variance”,cex.lab=1,font.lab=2,main = “Milk Urea” )<br>legend(0,8.5,bty=”0”, legend=c(“Parity1”,”Parity2”, “Parity3”),text.col=c(“blue”,”forestgreen”,”red”),<br>       col=c(“blue”,”forestgreen”,”red”),lty=c(1),cex=1.1,lwd=0.8, pch=c(1))</p>
<p>##total<br>par(font=2,las=1,mar = c(5,5,4,2) + 0.1)<br>plot(data$DIM, data$SDTOT.1, xlab =””, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”blue”,lty=1, xlim = c(5,365), ylim=c(min(data$SDTOT.1),max(data$SDTOT.2)))<br>par(new=TRUE)<br>plot(data$DIM, data$SDTOT.2, xlab =” “, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”forestgreen”,lty=1, xlim = c(5,365), ylim=c(min(data$SDTOT.1),max(data$SDTOT.2)))<br>par(new=TRUE)<br>plot(data$DIM, data$SDTOT.3, xlab =” “, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”red”,lty=1, xlim = c(5,365), ylim=c(min(data$SDTOT.1),max(data$SDTOT.2)))</p>
<p>minor.tick(nx=2, ny=2,tick.ratio=0.75)<br>par(mar = c(5,4,4,2) + 0.1)<br>title(xlab=”Days in Milk”,ylab=”Total variance”,cex.lab=1,font.lab=2,main = “Milk Urea” )<br>legend(0,60,bty=”0”, legend=c(“Parity1”,”Parity2”, “Parity3”),text.col=c(“blue”,”forestgreen”,”red”),<br>       col=c(“blue”,”forestgreen”,”red”),lty=c(1),cex=1.1,lwd=0.8, pch=c(1))</p>
<p>##h2<br>par(font=2,las=1,mar = c(5,5,4,2) + 0.1)<br>plot(data$DIM, data$SHER.1, xlab =””, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”blue”,lty=1, xlim = c(5,365), ylim=c(min(data$SHER.3),max(data$SHER.2)))<br>par(new=TRUE)<br>plot(data$DIM, data$SHER.2, xlab =” “, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”forestgreen”,lty=1, xlim = c(5,365), ylim=c(min(data$SHER.3),max(data$SHER.2)))<br>par(new=TRUE)<br>plot(data$DIM, data$SHER.3, xlab =” “, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”red”,lty=1, xlim = c(5,365), ylim=c(min(data$SHER.3),max(data$SHER.2)))</p>
<p>minor.tick(nx=2, ny=2,tick.ratio=0.75)<br>par(mar = c(5,4,4,1) + 0.1)<br>title(xlab=”Days in Milk”,ylab=”Heritabilities”,cex.lab=1,font.lab=2,main = “Milk Urea” )<br>legend(0,0.248,bty=”0”, legend=c(“Parity1”,”Parity2”, “Parity3”),text.col=c(“blue”,”forestgreen”,”red”),<br>       col=c(“blue”,”forestgreen”,”red”),lty=c(1),cex=1.1,lwd=0.8, pch=c(1))<br>#########cor<br>cor &lt;-  read.table(“MUcorre3RF.csv”,sep=”,”,header=T,fill=TRUE)<br>head(cor)</p>
<p>##genetic correlation<br>par(font=2,las=1,mar = c(5,5,4,2) + 0.1)<br>plot(cor$DIM, cor$Corr.GEN.1.2, xlab =””, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”blue”,lty=1, xlim = c(5,365), ylim=c(min(cor$Corr.GEN.1.3),max(cor$Corr.GEN.2.3)))<br>par(new=TRUE)<br>plot(cor$DIM, cor$Corr.GEN.1.3, xlab =” “, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”forestgreen”,lty=1, xlim = c(5,365), ylim=c(min(cor$Corr.GEN.1.3),max(cor$Corr.GEN.2.3)))<br>par(new=TRUE)<br>plot(cor$DIM, cor$Corr.GEN.2.3, xlab =” “, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”red”,lty=1, xlim = c(5,365), ylim=c(min(cor$Corr.GEN.1.3),max(cor$Corr.GEN.2.3)))</p>
<p>minor.tick(nx=2, ny=2,tick.ratio=0.75)<br>par(mar = c(5,4,4,2) + 0.1)<br>title(xlab=”Days in Milk”,ylab=”Genetic Correlation”,cex.lab=1,font.lab=2,main = “Milk Urea” )<br>legend(0,0.81,bty=”0”, legend=c(“Parity1-Parity2”,”Parity1-Parity3”, “Parity2-Parity3”),text.col=c(“blue”,”forestgreen”,”red”),<br>       col=c(“blue”,”forestgreen”,”red”),lty=c(1),cex=1.1,lwd=0.8, pch=c(1))</p>
<p>#phe correlation<br>par(font=2,las=1,mar = c(5,5,4,2) + 0.1)<br>plot(cor$DIM, cor$Corr.PHENO.1.2, xlab =””, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”blue”,lty=1, xlim = c(5,365), ylim=c(min(cor$Corr.PHENO.1.3),max(cor$Corr.PHENO.1.2)))<br>par(new=TRUE)<br>plot(cor$DIM, cor$Corr.PHENO.1.3, xlab =” “, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”forestgreen”,lty=1, xlim = c(5,365), ylim=c(min(cor$Corr.PHENO.1.3),max(cor$Corr.PHENO.1.2)))<br>par(new=TRUE)<br>plot(cor$DIM, cor$Corr.PHENO.2.3, xlab =” “, ylab = “” ,type=”o”,lwd=0.8,<br>     col=”red”,lty=1, xlim = c(5,365), ylim=c(min(cor$Corr.PHENO.1.3),max(cor$Corr.PHENO.1.2)))</p>
<p>minor.tick(nx=2, ny=2,tick.ratio=0.75)<br>par(mar = c(5,4,4,2) + 0.1)<br>title(xlab=”Days in Milk”,ylab=”Phenotype Correlation”,cex.lab=1,font.lab=2,main = “Milk Urea” )<br>legend(0,0.55,bty=”0”, legend=c(“Parity1-Parity2”,”Parity1-Parity3”, “Parity2-Parity3”),text.col=c(“blue”,”forestgreen”,”red”),<br>       col=c(“blue”,”forestgreen”,”red”),lty=c(1),cex=1.1,lwd=0.8, pch=c(1))</p>
<p>##############################分屏幕</p>
<p>split.screen(c(3,2))<br>screen(1) #屏幕1预备输出<br>plot(cor$DIM, cor$Corr.GEN.1.2, xlab =”Days in Milk”, ylab = “Genetic Correlations” ,type=”o”,lwd=0.8,<br>     col=”darkgreen”,lty=1, xlim = c(5,365), main = “Parity1 Fat and Parity2 Fat”)<br>screen(2) #屏幕2预备输出<br>plot(cor$DIM, cor$Corr.PHENO.1.2, xlab =”Days in Milk”, ylab = “Phenotype Correlations” ,type=”o”,lwd=0.8,<br>     col=”darkgreen”,lty=1, xlim = c(5,365), main = “ Parity1 Fat and Parity2 Fat”)</p>
<p>screen(3) #屏幕1预备输出<br>plot(cor$DIM, cor$Corr.GEN.1.3, xlab =”Days in Milk”, ylab = “Genetic Correlations” ,type=”o”,lwd=0.8,<br>     col=”darkblue”,lty=1, xlim = c(5,365), main = “Parity1 Fat and Parity3 Fat”)<br>screen(4) #屏幕2预备输出<br>plot(cor$DIM, cor$Corr.PHENO.1.3, xlab =”Days in Milk”, ylab = “Phenotype Correlations” ,type=”o”,lwd=0.8,<br>     col=”darkblue”,lty=1, xlim = c(5,365), main = “Parity1 Fat and Parity3 Fat”)</p>
<p>screen(5) #屏幕1预备输出<br>plot(cor$DIM, cor$Corr.GEN.2.3, xlab =”Days in Milk”, ylab = “Genetic Correlations” ,type=”o”,lwd=0.8,<br>     col=”darkgreen”,lty=1, xlim = c(5,365), main = “Parity2 Fat and Parity3 Fat”)<br>screen(6) #屏幕2预备输出<br>plot(cor$DIM, cor$Corr.PHENO.2.3, xlab =”Days in Milk”, ylab = “Phenotype Correlations” ,type=”o”,lwd=0.8,<br>     col=”darkgreen”,lty=1, xlim = c(5,365), main = “Parity2 Fat and Parity3 Fat”)</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="2020/03/12/美国签证/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="images/avatar.gif">
      <meta itemprop="name" content="Yansen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Animal genetic">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="2020/03/12/%E7%BE%8E%E5%9B%BD%E7%AD%BE%E8%AF%81/" target="_blank" rel="noopener" class="post-title-link" itemprop="url">中国人在比利时申请美国签证</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">نُشر في</span>
              

              <time title="أُنشأ: 2020-03-12 19:24:55 / عُدل: 20:00:14" itemprop="dateCreated datePublished" datetime="2020-03-12T19:24:55+01:00">2020-03-12</time>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>先说下本人情况，一名比利时某大学的在读博士。计划今年五月去美国某大学培训3周的技能课。<br>申请：B1/B2（有这个专门的选项） 签证</p>
<p>首先填写：DS-160表格： <a href="https://ceac.state.gov/genniv/" target="_blank" rel="noopener">https://ceac.state.gov/genniv/</a>   ； 网上有很多教程，我就不多写了，实际上按自己的实际信息给就可以了（其中还有父母的生日，可怕！！），需要大概1个多小时完成。<br>填完记得打印，并且发到自己邮箱备份。  ！！！这个非常重要</p>
<p>完成后，就可以交钱预约面谈了： <a href="https://ais.usvisa-info.com/en-be/niv" target="_blank" rel="noopener">https://ais.usvisa-info.com/en-be/niv</a>   ；这个只适合在比利时申请的网址。<br>首先创建账号；按DS-160相同信息填写。<br>再进入，填写个人信息。 交费：152€；  选择预约时间，只能选1周后（从交钱开始）。一般都是只有上午。</p>
<p>去面签前需准备的材料：<br>1 确认的DS-160表格<br>2 一个5cmx5cm照片（近半年照的）<br>3 预约成功的表格（没用到）</p>
<p>按预约的时间准时到达（至多提前15分钟），拿出自己护照，就会让你进入，PS：不要带电脑。手机、包其他的可以。 进入第二次安检时，可以寄存。</p>
<p>安检完后，一进门，就是需要排队进行资格审查的窗口（3个），审查完后，就去隔壁，排队面签。</p>
<p>本人面签中问了一下问题：<br>1 去美国干什么？呆多久？<br>2 要去美国那里？<br>3 完成硕士学位没？<br>4 自己博士和硕士专业各是什么？<br>5 对方学校是否给本人可以注册的证明？（没有。。。。，因为本人不需要）</p>
<p>有两个人面试，一个是专门英文的，一个是英文和法语均可以的。推荐只说英语的。<br>因为本人面试了两次，第一次那人说听不懂（本人也是心里苦）， 隔了半小时，又去了一次，通过了。</p>
<p>周一上午面试， 周三中午拿到自己的护照。相当神速，而且还是十年的签证。</p>
<p>总来的说，这个签证，速度真的非常牛，从开始，到拿到，一共10天。<br>对了，拿到后要去：<a href="https://www.evus.gov/#/" target="_blank" rel="noopener">https://www.evus.gov/#/</a>  这里进行注册，两年需要更新一次。</p>
<p>希望可以帮到你哈！</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="2020/03/10/liunx-2/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="images/avatar.gif">
      <meta itemprop="name" content="Yansen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Animal genetic">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="2020/03/10/liunx-2/" target="_blank" rel="noopener" class="post-title-link" itemprop="url">Liunx(2)&BLUPF90</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">نُشر في</span>
              

              <time title="أُنشأ: 2020-03-10 21:41:34 / عُدل: 21:42:55" itemprop="dateCreated datePublished" datetime="2020-03-10T21:41:34+01:00">2020-03-10</time>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>查看大的文件：<br>less –S file<br>wc -l genotype.txt pedigree.txt</p>
<p>expand             replaces TAB with spaces<br>uniq                    retains unique lines on a sorted file</p>
<blockquote>
<blockquote>
<pre><code>代表为在file后面添加数据</code></pre><p>Cat 是以列相叠加<br>cat file1 file2 &gt; output_file<br>Pase 以行进行连接，但是以TAB连接，需要使用expand处理，其会使tab变成空格<br>Eg: pase fileA fileB | expand | combind C<br>但是：paste –d “ “     可直接以空格相连接                   </p>
</blockquote>
</blockquote>
<p>Awk：<br>Implicit variables<br>NF - number of fields<br>NR - record number<br>FS - input field separator<br>OFS - output field separator<br>• Print columns 1, 2, and last of solutions file<br>awk ‘{print $1,$2,$NF}’ solutions &gt; sol_out<br>• Print all the columns:<br>awk ‘{print $0}’ solutions &gt; sol_out<br>• Print columns 3 and 4 based on occurrence in column 2:<br>awk ‘{if ($2==2) print $3,$4}’ solutions &gt; sol_out<br>• Print columns 3 and 4 skipping the first line:<br>awk ‘{if (NR&gt;1) print $3,$4}’ solutions &gt; sol_out<br>• Print length of column 2 from line 1:<br>awk ‘{if (NR==1) print length($2)}’ snp.txt<br>• Process CSV files<br>awk ‘BEGIN {FS=”,”} {print $1,$2,$3}’ pedigree.txt &gt; ped_out</p>
<p>看每个父亲有多少后代：<br>awk ‘{ sire[$2]+=1} END { for (s in sire)<br>         {print “Sire “ s, sire[s]}}’ pedigree.txt</p>
<p>或者（借助uniq）：awk ‘$2&gt;0{ print $2}’ ped | sort | uniq –c &gt; s.temp</p>
<p>对于BLUPF90的基因文件的SNP数统计：<br>awk ‘{print length($2)}’ snp.txt</p>
<p>每个人都具有相同数量的SNP基因型：<br>awk ‘{print length($2)}’ snp.txt | sort –u</p>
<p>提取前200个个体的SNP:<br>awk ‘NR &lt;= 200’ snp.txt &gt; anim200.temp</p>
<p>BLUPF90软件的一些学习：</p>
<p>PreGSf90:对于map的要求（新形式）：<br>OPTION map_info  <file><br>–SNPID for SNP<br>– CHR for chromosome<br>– POS for position</p>
<p>• How to do only QC avoiding the inverses:只进行QC<br>– OPTION SNP_file  marker.geno<br>– OPTION saveCleanSNPs<br>– OPTION createGInverse 0<br>– OPTION createA22Inverse 0<br>– OPTION createGimA22i 0<br>但是只想运行：使用以下选项：<br>• OPTION SNP_file marker.geno_clean<br>• OPTION no_quality_control</p>
<p>计算G矩阵：</p>
<p>• OPTION maxsnp x  设定SNP的最大数<br>– Set the maximum length of string for reading marker<br>data from file =&gt; BovineHD chip</p>
<p>OPTION AlphaBeta alpha beta<br>– G = alpha<em>G + beta</em>A<br>• OPTION tunedG x<br>– 0: no adjustment<br>– 1: mean(diag(G))=1, mean(offdiag(G))=0<br>– 2: mean(diag(G))=mean(diag(A)), mean(offdiag(G))=mean(offdiag(A)) (default)<br>– 3: mean(G)=mean(A)<br>– 4: Use Fast adjustment.<br>OPTION saveG [all]<br>– If the optional all is present all intermediate G matrices will be saved!!! Or it inverted</p>
<p>OPTION saveGInverse<br>– Only the final matrix G, after blending, scaling, etc. is inverted !!!</p>
<p> Population structure:<br>OPTION plotpca</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="2020/02/26/annovar使用/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="images/avatar.gif">
      <meta itemprop="name" content="Yansen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Animal genetic">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="2020/02/26/annovar%E4%BD%BF%E7%94%A8/" target="_blank" rel="noopener" class="post-title-link" itemprop="url">annovar注释软件</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">نُشر في</span>
              

              <time title="أُنشأ: 2020-02-26 22:32:56 / عُدل: 22:48:27" itemprop="dateCreated datePublished" datetime="2020-02-26T22:32:56+01:00">2020-02-26</time>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>##annovar解压包<br>ANNOVAR<br>│  annotate_variation.pl #主程序，功能包括下载数据库，三种不同的注释<br>│  coding_change.pl #可用来推断蛋白质序列<br>│  convert2annovar.pl #将多种格式转为.avinput的程序<br>│  retrieve_seq_from_fasta.pl #用于自行建立其他物种的转录本<br>│  table_annovar.pl #注释程序，可一次性完成三种类型的注释<br>│  variants_reduction.pl #可用来更灵活地定制过滤注释流程<br>│<br>├─example #存放示例文件<br>│<br>└─humandb #人类注释数据库</p>
<p>##数据库下载<br>$ Perl annotate_variation.pl -buildver hg19 -downdb -webfrom annovar refGene humandb/<br> -buildver 表示version<br> -downdb 下载数据库的指令<br> -webfrom annovar 从annovar提供的镜像下载，不加此参数将寻找数据库本身的源<br> humandb/ 存放于humandb/目录下</p>
<p>##输入的文件格式<br>NNOVAR使用.avinput格式，如以上代码所示，该格式每列以tab分割，需要有以下几个信息：<br>染色体位置 起始位点 终止位点 参考基因组碱基 突变碱基 ……</p>
<p>##ANNOVAR输入文件的格式转换</p>
<p>ANNOVAR主要使用convert2annovar.pl程序进行转换，转换后文件是精简过的，主要包含前面提到的5列内容，如果要将原格式的文件的所有内容都包含在转换后的.avinput文件中，可以使用-includeinfo参数；如果需要分开每个sample输出单一的.avinput文件，可以使用-allsample参数，等等。 代码示例：<br>$ convert2annovar.pl -format vcf4 example/ex2.vcf &gt; ex2.avinput</p>
<h1 id="format-vcf4-指定格式为vcf"><a href="#format-vcf4-指定格式为vcf" target="_blank" rel="noopener" class="headerlink" title="-format vcf4 指定格式为vcf"></a>-format vcf4 指定格式为vcf</h1><p>ANNOVAR还主要支持以下格式转换：<br>SAMtools pileup format Complete Genomics format GFF3-SOLiD calling format SOAPsnp calling format MAQ calling format CASAVA calling format</p>
<h3 id="使用table-annovar-pl注释多个数据库"><a href="#使用table-annovar-pl注释多个数据库" target="_blank" rel="noopener" class="headerlink" title="使用table_annovar.pl注释多个数据库"></a>使用table_annovar.pl注释多个数据库</h3><p>perl table_annovar.pl example/ex1.avinput humandb/ -buildver hg19 -out myanno -remove -protocol refGene,cytoBand,exac03,avsnp147,dbnsfp30a -operation gx,r,f,f,f -nastring . -csvout -polish -xref example/gene_xref.txt<br>#-bulidver hg19 表示使用的参考基因组版本<br>#-out myanno 表示输出文件前缀，亦可用 –outfile 直接指定文件名<br>#-remove 表示删除中间文件<br>#-protocol 表示使用的数据库，其数据库顺序要与后面的operation注释方式对应上<br>#-operation 表示对应数据库的注释类型（g代表gene-based、r代表region-based、f代表filter-based，gx means gene-based with cross-reference annotation (from -xref argument)）<br>#-nasting . 点号代替缺省值<br>#-csvout 表示输出为csv格式</p>
<p>Table_annovar.pl（可一次完成三种类型的注释）<br>使用ANNOVAR最简单的方法就是使用table_annovar.pl进行注释，它的输入文件可以是多种格式包括VCF，输出文件已Tab分隔，每一列代表着一种注释。 注释命令示例：</p>
<p>$ANNOVAR/annovar/table_annovar.pl 15_indel_pre.avinput.hg19.variant2.avinput ~/ANNOVAR/annovar/humandb/ -buildver hg19 -out myanno -remove -protocol refGene,cytoBand,genomicSuperDups,esp6500siv2_all,1000g2015aug_all,1000g2015aug_eur,exac03,avsnp147,dbnsfp30a -operation g,r,r,f,f,f,f,f,f -nastring . -csvout</p>
<h1 id="buildver-hg19-表示使用hg19版本"><a href="#buildver-hg19-表示使用hg19版本" target="_blank" rel="noopener" class="headerlink" title="-buildver hg19 表示使用hg19版本"></a>-buildver hg19 表示使用hg19版本</h1><h1 id="out-myanno-表示输出文件的前缀为myanno"><a href="#out-myanno-表示输出文件的前缀为myanno" target="_blank" rel="noopener" class="headerlink" title="-out myanno 表示输出文件的前缀为myanno"></a>-out myanno 表示输出文件的前缀为myanno</h1><h1 id="remove-表示删除注释过程中的临时文件"><a href="#remove-表示删除注释过程中的临时文件" target="_blank" rel="noopener" class="headerlink" title="-remove 表示删除注释过程中的临时文件"></a>-remove 表示删除注释过程中的临时文件</h1><h1 id="protocol-表示注释使用的数据库，用逗号隔开，且要注意顺序"><a href="#protocol-表示注释使用的数据库，用逗号隔开，且要注意顺序" target="_blank" rel="noopener" class="headerlink" title="-protocol 表示注释使用的数据库，用逗号隔开，且要注意顺序"></a>-protocol 表示注释使用的数据库，用逗号隔开，且要注意顺序</h1><h1 id="operation-表示对应顺序的数据库的类型（g代表gene-based、r代表region-based、f代表filter-based），用逗号隔开，注意顺序"><a href="#operation-表示对应顺序的数据库的类型（g代表gene-based、r代表region-based、f代表filter-based），用逗号隔开，注意顺序" target="_blank" rel="noopener" class="headerlink" title="-operation 表示对应顺序的数据库的类型（g代表gene-based、r代表region-based、f代表filter-based），用逗号隔开，注意顺序"></a>-operation 表示对应顺序的数据库的类型（g代表gene-based、r代表region-based、f代表filter-based），用逗号隔开，注意顺序</h1><h1 id="nastring-表示用点号替代缺省的值"><a href="#nastring-表示用点号替代缺省的值" target="_blank" rel="noopener" class="headerlink" title="-nastring . 表示用点号替代缺省的值"></a>-nastring . 表示用点号替代缺省的值</h1><h1 id="csvout-表示最后输出-csv文件"><a href="#csvout-表示最后输出-csv文件" target="_blank" rel="noopener" class="headerlink" title="-csvout 表示最后输出.csv文件"></a>-csvout 表示最后输出.csv文件</h1><p>输出的csv文件将包含输入的5列主要信息以及各个数据库里的注释，此外，table_annoval.pl可以直接对vcf文件进行注释（不需要转换格式），<br>注释的内容将会放在vcf文件的“INFO”那一栏。</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="2020/02/26/GLM/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="images/avatar.gif">
      <meta itemprop="name" content="Yansen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Animal genetic">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="2020/02/26/GLM/" target="_blank" rel="noopener" class="post-title-link" itemprop="url">GLM</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">نُشر في</span>

              <time title="أُنشأ: 2020-02-26 11:31:00" itemprop="dateCreated datePublished" datetime="2020-02-26T11:31:00+01:00">2020-02-26</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">عُدل في</span>
                <time title="عُدل: 2020-05-26 08:22:51" itemprop="dateModified" datetime="2020-05-26T08:22:51+02:00">2020-05-26</time>
              </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>前言<br>之前我们使用R进行线性分析时，我们得到了满足正态分布下的线性拟合模型，但在许多实际情况中，假定满足正态分布的假设并不合理<br>结果变量为类别型：包括二值变量（是/否、通过/失败、存活/死亡）和多类别变量（优/良/可/差）等。<br>结果变量为计数型的（一周交通事故的数目、每日酒水消耗的数量），这类变量都是非负的有限值，它们的均值和方差通常都是相关的（而正态分布变量间则是相互独立）。<br>为此，我们在一般线性的基础上进行深入学习，来研究包含了非正态因变量的广义线性模型。</p>
<p>一、GLM函数概况<br>在R语言中，我们使用glm()函数构建广义线性模型，调用语法如下：</p>
<h1 id="family为拟合所属的函数族"><a href="#family为拟合所属的函数族" target="_blank" rel="noopener" class="headerlink" title="family为拟合所属的函数族"></a>family为拟合所属的函数族</h1><h1 id="function为对应的连接函数"><a href="#function为对应的连接函数" target="_blank" rel="noopener" class="headerlink" title="function为对应的连接函数"></a>function为对应的连接函数</h1><p>glm(formula,family=family(link = function),data=)<br>常用的分布族和连接函数见下表：</p>
<ol>
<li>logistic回归适用于二值响应变量，连接函数为logit函数，概率分布为二项分布：</li>
</ol>
<p>glm(Y ~ X1 + X2 + X3, family=binomial(link=”logit”), data=data)<br>2. 泊松回归适用于在给定时间内响应变量为事件发生数目的情形，其假设Y服从泊松分布，连接函数为log(λ)，概率分布为泊松分布：</p>
<p>glm(Y ~ X1 + X2 + X3, family=poisson(link=”log”), data=data)<br>除基本公式外，广义线性模型还会用到其他辅助函数，包括summary()，coef()等，这些函数的使用和线性模型中lm()的配套辅助函数一致，具体可参阅：R Language Learning：线性回归（五）</p>
<p>总之，广义线性模型通过拟合响应变量的条件均值的一个函数（不是响应变量的条件均值），并假设响应变量服从指数分布族中的某个分布（不限于正态分布），从而极大地扩展了标准线性模型。模型参数估计的推导依据是极大似然估计，而非最小二乘法。</p>
<p>二、Logistics回归分析<br>当因变量为二分类或多分类时，Logistics回归是非常重要的模型。由于Logistics回归对资料的正态性和方差齐性不做要求、对自变量类型也不做要求，使得Logistic回归模型在众多领域被广泛使用。<br>例如，想探讨胃癌发生的危险因素，可以选择两组人群，一组是胃癌组，一组是非胃癌组，两组人群肯定有不同的体征和生活方式等。这里的因变量为是否胃癌，即“是”或“否”，为两分类变量，自变量就可以包括很多了，例如年龄、性别、饮食习惯等。自变量既可以是连续的，也可以是分类的。通过logistic回归分析，就可以大致了解到底哪些因素是胃癌的危险因素。<br>估计的Logistics回归方程为：</p>
<p>[公式] or [公式]</p>
<p>Logistics回归模型的表达形式为：[公式]</p>
<p>模型解读：[公式]为暴露于某种状态下的结局概率，[公式]是一种变量变换方式，表示对[公式]进行[公式]变换，[公式]为偏回归系数，表示在其他自变量不变的条件下，每变化一个单位[公式]的估计值。</p>
<p>Logistics回归是通过最大似然估计求解常数项和偏回归系数，基本思想时当从总体中随机抽取n个样本后，最合理的参数估计量应该使得这n个样本观测值的概率最大。最大似然法的基本思想是先建立似然函数与对数似然函数，再通过使对数似然函数最大求解相应的参数值，所得到的估计值称为参数的最大似然估计值。</p>
<ol>
<li>数据准备（类别型变量进行0/1量化）</li>
</ol>
<p>首先，我们选用AER包中的Affairs数据集来构建Logistics回归模型，这个数据集记录了一组婚外情数据，其中包括参与者性别、年龄、婚龄、是否有小孩、宗教信仰程度（5分制，1表示反对，5表示非常信仰）、学历、职业和婚姻的自我评分（5分制，1表示非常不幸福，5表示非常幸福）。</p>
<p>在使用数据集之前，载入AER包</p>
<blockquote>
<p>library(AER)<br>data(Affairs,package=”AER”)<br>对于这个数据集，我们关注是否出轨，即这是一个二值型结果（出轨过/从未出轨）。因此，我们接下来将’affaris’特征转化为二值型因子’ynaffair’，该二值型因子即可以作为Logistic回归的结果变量。<br>Affairs$ynaffair[Affairs$affairs &gt; 0] &lt;- 1<br>Affairs$ynaffair[Affairs$affairs== 0] &lt;- 0<br>Affairs$ynaffair &lt;-factor(Affairs$ynaffair,levels=c(0,1),labels=c(“No”,”Yes”))</p>
</blockquote>
<ol start="2">
<li>Logistics模型构建：</li>
</ol>
<blockquote>
<p>myfit &lt;- glm(ynaffair ~ gender + age + yearsmarried + children + religiousness + education + occupation + rating, data=Affairs, family=binomial())<br>summary(myfit)<br>Coefficients:<br>              Estimate Std. Error z value Pr(&gt;|z|)<br>(Intercept)    1.37726    0.88776   1.551 0.120807<br>gendermale     0.28029    0.23909   1.172 0.241083<br>age           -0.04426    0.01825  -2.425 0.015301 *<br>yearsmarried   0.09477    0.03221   2.942 0.003262 **<br>childrenyes    0.39767    0.29151   1.364 0.172508<br>religiousness -0.32472    0.08975  -3.618 0.000297 <strong>*<br>education      0.02105    0.05051   0.417 0.676851<br>occupation     0.03092    0.07178   0.431 0.666630<br>rating        -0.46845    0.09091  -5.153 2.56e-07 *</strong></p>
</blockquote>
<hr>
<p>Signif. codes:  0 ‘<strong>*’ 0.001 ‘</strong>’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1<br>从回归系数的p值（最后一栏）可以看到，性别、是否有孩子、学历和职业对方程的贡献都不显著（p值较大）。去除这些变量重新拟合模型，检验新模型是否拟合得更好。</p>
<blockquote>
<p>myfit_reduced &lt;- glm(ynaffair ~ age + yearsmarried + religiousness + rating, data=Affairs, family=binomial())<br>summary(myfit_reduced)<br>Coefficients:<br>              Estimate Std. Error z value Pr(&gt;|z|)<br>(Intercept)    1.93083    0.61032   3.164 0.001558 **<br>age           -0.03527    0.01736  -2.032 0.042127 *<br>yearsmarried   0.10062    0.02921   3.445 0.000571 <strong>*<br>religiousness -0.32902    0.08945  -3.678 0.000235 *</strong><br>rating        -0.46136    0.08884  -5.193 2.06e-07 ***<br>从最后一列可以看到，新模型的每个回归系数都非常显著（p&lt;0.05）</p>
</blockquote>
<ol start="3">
<li>多模型比较</li>
</ol>
<p>由于这个两模型是嵌套关系，所以我们可以使用anova()函数对它们进行比较。而对于广义线性回归，可用卡方检验进行判断。</p>
<blockquote>
<p>anova(myfit,myfit_reduced,test=”Chisq”)<br>Analysis of Deviance Table<br>Model 1: ynaffair ~ gender + age + yearsmarried + children + religiousness +<br>    education + occupation + rating<br>Model 2: ynaffair ~ age + yearsmarried + religiousness + rating<br>  Resid. Df Resid. Dev Df Deviance Pr(&gt;Chi)<br>1       592     609.51<br>2       596     615.36 -4  -5.8474   0.2108<br>检验结果的卡方值不显著（p=0.21），这表明，四个预测变量的新模型和九个预测变量的旧模型模拟程度其实差不多。一般情况下我们更倾向于选择四个预测变量的新模型。</p>
</blockquote>
<ol start="4">
<li>评价预测变量对结果概率的影响（predict函数）</li>
</ol>
<p>既然已经有了模型，那么给定一个用户行为，预测其出轨的概率。可以使用predict()函数进行预测，以下代码生成了一些虚拟数据并进行了预测，主要评测婚姻评分对出轨概率的影响。</p>
<h1 id="构建dataframe"><a href="#构建dataframe" target="_blank" rel="noopener" class="headerlink" title="构建dataframe"></a>构建dataframe</h1><blockquote>
<p>testdata &lt;- data.frame(rating=c(1, 2, 3, 4, 5), age=mean(Affairs$age), yearsmarried=mean(Affairs$yearsmarried), religiousness=mean(Affairs$religiousness))</p>
</blockquote>
<h1 id="添加一列为概率预测值"><a href="#添加一列为概率预测值" target="_blank" rel="noopener" class="headerlink" title="添加一列为概率预测值"></a>添加一列为概率预测值</h1><blockquote>
<p>testdata$prob &lt;- predict(myfit_reduced, newdata=testdata, type=”response”)<br>testdata<br>  rating      age yearsmarried religiousness      prob<br>1      1 32.48752     8.177696      3.116473 0.5302296<br>2      2 32.48752     8.177696      3.116473 0.4157377<br>3      3 32.48752     8.177696      3.116473 0.3096712<br>4      4 32.48752     8.177696      3.116473 0.2204547<br>5      5 32.48752     8.177696      3.116473 0.1513079<br>从结果可以看到，当婚姻评分从1分（很不幸福）增加到5分（非常幸福）时，出轨概率从0.53降低至0.15。</p>
</blockquote>
<blockquote>
<p>testdata &lt;- data.frame(rating=mean(Affairs$rating), age=seq(17, 57, 10), yearsmarried=mean(Affairs$yearsmarried), religiousness=mean(Affairs$religiousness))<br>testdata$prob &lt;- predict(myfit_reduced, newdata=testdata, type=”response”)<br>testdata<br>   rating age yearsmarried religiousness      prob<br>1 3.93178  17     8.177696      3.116473 0.3350834<br>2 3.93178  27     8.177696      3.116473 0.2615373<br>3 3.93178  37     8.177696      3.116473 0.1992953<br>4 3.93178  47     8.177696      3.116473 0.1488796<br>5 3.93178  57     8.177696      3.116473 0.1094738<br>在年龄的影响方面，可以看出，当其他变量不变、年龄从17增加到57时，出轨的概率从0.34降低至0.11。</p>
</blockquote>
<ol start="5">
<li>过度离势</li>
</ol>
<p>过度离势指的是观测到的响应变量方差大于期望的二项分布方差，过度离势会导致奇异的标准误检验和不精确的显著性检验。</p>
<p>当出现过度离势时，仍可使用glm()函数拟合Logistic回归，但此时需要将二项分布改为类二项分布。</p>
<p>检测过度离势的一种方法是比较二项分布模型的残差偏差与残差自由度（在四变量模型中分别是615.36和596），如果两者的比值比1大很多，则可以认为存在过度离势。</p>
<ol start="6">
<li>Logistic回归模型的进阶学习</li>
</ol>
<p>扩展的Logistic回归和变种如下所示：</p>
<p>稳健Logistic回归：当拟合Logistic回归模型数据出现离群点和强影响点时，robust包中的glmRob()函数可用来拟合稳健的广义线性模型<br>多项分布回归：当响应变量包含两个以上的无序类别（例如已婚、寡居、离婚）时，可使用mlogit包中的mlogit()函数拟合多项Logistic回归<br>序数Logistic回归：当响应变量是一组有序的类别（例如信用为好/良/差）时，可使用rms包中的lrm()函数拟合序数Logistic回归。</p>
<p>三、泊松回归<br>泊松回归的自变量是连续性或类别型变量，因变量是计数型的变量。泊松回归因变量通常局限在一个固定长度时间段内进行测量（如过去一年交通事故数），整个观测集中时间长度都是不变的。Poisson回归主要有两个假设，首先，具有相同特征和同时的不同对象的人时风险是同质的，其次，当样本量越来越大时，频数的均数趋近于方差。<br>调用模型的公式如下：</p>
<blockquote>
<p>myfit &lt;- glm(y~x1+x2+…+xn,data=,family = poisson)</p>
</blockquote>
<ol>
<li>数据准备</li>
</ol>
<p>robust包中Breslow癫痫数据记录了治疗初期八周内，抗癫痫药物对癫痫发病数的影响。响应变量为sumY（随机化后八周内癫痫发病数），预测变量为治疗条件（Trt）、年龄（Age）和前八周内的基础癫痫发病数（Base），在这个数据集中，我们感兴趣的是药物治疗能否减少癫痫发病数。</p>
<blockquote>
<p>data(breslow.dat, package=”robust”)<br>接下来我们考察响应变量，基础和随机化后的癫痫发病数都有很高的偏度，从图中可以清楚地看到因变量的偏倚特性以及可能的离群点。</p>
</blockquote>
<blockquote>
<p>library(ggplot2)<br>p1 &lt;- ggplot(breslow.dat,aes(sumY)) + geom_histogram(color=’green’)<br>p2 &lt;- ggplot(breslow.dat,aes(Trt,sumY)) + geom_bar()</p>
</blockquote>
<h1 id="组合图形"><a href="#组合图形" target="_blank" rel="noopener" class="headerlink" title="组合图形"></a>组合图形</h1><blockquote>
<p>library(gridExtra)<br>grid.arrange(p1,p2,nrow=1)</p>
</blockquote>
<p>从图中可以清楚的看到因变量的偏移特性及可能的离群点。药物治疗下癫痫的发病数似乎变小，且方差也变小了。</p>
<ol start="2">
<li>构建模型</li>
</ol>
<blockquote>
<p>myfit &lt;- glm(sumY ~ Base + Age + Trt, data=breslow.dat, family=poisson())<br>summary(myfit)</p>
</blockquote>
<ol start="3">
<li>过度离势</li>
</ol>
<p>在泊松分布中同样可能存在过度离势，泊松分布的方差和均值相等，当响应变量观测的方差比依据泊松分布预测的方差大时，可能存在过度离势。</p>
<blockquote>
<p>summary(myfit)</p>
</blockquote>
<p>(Dispersion parameter for poisson family taken to be 1)</p>
<pre><code>Null deviance: 2122.73  on 58  degrees of freedom</code></pre><p>Residual deviance:  559.44  on 55  degrees of freedom<br>AIC: 850.71</p>
<p>Number of Fisher Scoring iterations: 5<br>与Logistic回归类似，但此例中残差偏差（559.44）与残差自由度（55）的比例为10.17，这一值远大于1，故存在过度离势。在这种情况下，可以考虑使用类泊松回归替代泊松回归（family=quasipoisson()）。</p>
<p>除此之外，我们还可以使用qcc包来检验泊松模型是否存在过度离势。</p>
<blockquote>
<p>library(qcc)<br>qcc.overdispersion.test(breslow.dat$sumY, type=”poisson”)</p>
</blockquote>
<ol start="4">
<li>泊松回归模型的进阶学习：</li>
</ol>
<p>时间段变化的泊松回归：如果每个观测的时间段长短不同，则因变量需要从计数更改为比率，即发生次数除以观测时间，使用glm()的offset选项即可，offset=log(time)，其中time是每名病人的观测时间<br>零膨胀的泊松回归：在一个数据集中，0计数的数目时常比用泊松模型预测得多。在Affairs数据集中，很有可能有一群从未出轨的对象，他们称为数据集的结构零值。可以用零膨胀的泊松回归来分析这种情况，它将同时拟合两个模型，一个用来预测哪些人又会出轨，另一个用来预测排除了结构零值之后的调查对象会出轨多少次。可以将其看作是Logistic回归（预测结构零值）和泊松回归（预测无结构零值观测的计数）的组合，pscl包的zeroinfl()函数可做零膨胀泊松回归<br>稳健泊松回归：robust包中的glmRob()函数可以拟合稳健广义线性模型，包括稳健泊松回归。</p>
<p>References：</p>
<p>R语言实战（第2版） (豆瓣)<br>R学习笔记－11 广义线型模型<br>Generalized Linear Models<br>RPubs - 广义线性模型<br>What does a generalized linear model do?</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="2020/02/25/R实现简单随机分组/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="images/avatar.gif">
      <meta itemprop="name" content="Yansen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Animal genetic">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="2020/02/25/R%E5%AE%9E%E7%8E%B0%E7%AE%80%E5%8D%95%E9%9A%8F%E6%9C%BA%E5%88%86%E7%BB%84/" target="_blank" rel="noopener" class="post-title-link" itemprop="url">R实现简单分组</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">نُشر في</span>
              

              <time title="أُنشأ: 2020-02-25 08:25:50 / عُدل: 08:28:33" itemprop="dateCreated datePublished" datetime="2020-02-25T08:25:50+01:00">2020-02-25</time>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>现在有100个数，想随机取其中31个作为组A，23个作为组B，17个作为组C，剩下的作为组D<br>问题：<br>解决一：<br>dat&lt;-rnorm(100);<br>index&lt;-sample(1:100,size=100);<br>a31&lt;-dat[index[1:31]];<br>b23&lt;-dat[index[32:54]];<br>c17&lt;-dat[index[55:100]];</p>
<p>或者<br>split(1:100, sample(rep(1:4, c(31, 23, 17, 29))))</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="2020/02/24/awk处理表型3SD/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="images/avatar.gif">
      <meta itemprop="name" content="Yansen">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Animal genetic">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="2020/02/24/awk%E5%A4%84%E7%90%86%E8%A1%A8%E5%9E%8B3SD/" target="_blank" rel="noopener" class="post-title-link" itemprop="url">Liunx-awk处理表型3SD</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">نُشر في</span>

              <time title="أُنشأ: 2020-02-24 16:06:00" itemprop="dateCreated datePublished" datetime="2020-02-24T16:06:00+01:00">2020-02-24</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">عُدل في</span>
                <time title="عُدل: 2020-05-23 22:07:27" itemprop="dateModified" datetime="2020-05-23T22:07:27+02:00">2020-05-23</time>
              </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>sd=$ (awk ‘{sum+=$3;sumsq+=$3<em>$3} END {print sqrt(sumsq/NR - (sum/NR)^2)</em>(NR/(NR-1))}’ datal.temp)</p>
<p>media=$ (awk ‘{sum+=$3} END {print (sum/NR)}’ data1.tmep)</p>
<p>awk -v media=”$media” - v sd=”$sd” ‘{if($3&gt;=media-3<em>$sd &amp;&amp; $3&lt;=$media+3</em>$media+3*$sd) print $1,$2,$3,1}’ data1.temp &gt; data2.temp</p>
<p>sed ‘s/&lt;0&gt;/99/‘ data2.temp &gt; data.txt</p>
<p>####删除文件头<br>sed -i ‘1 d’ ped.txt</p>
<p>#####################SNP 不同编码的处理<br>awk ‘{print $2}’ snp_orangal | tr 3 1| tr 4 1 | tr 0 9 | tr 2 0| tr 9 2| &gt; snp_gen.temp</p>
<p>##ID 提取<br>awk ‘{print $1}’ snp_orangal &gt; snp_id.temp</p>
<p>##ID 与snp 合并<br>psate -d” “ snp_id.temp snp_gen.temp &gt; snp1.temp</p>
<p>expand snp1.temp &gt; snp.txt</p>
<p>#删去标头<br>sed -i ‘1 d’ snp.txt</p>
<p>##看文件<br>You can use cat to print large files and then scroll through the output, but it is usually more convenient to page the output. The original command for doing this was called more, but it has been superseded by a more powerful command called less. (This kind of naming is what passes for humor in the Unix world.) When you less a file, one page is displayed at a time; you can press spacebar to page down or type q to quit.</p>
<p>If you give less the names of several files, you can type :n (colon and a lower-case ‘n’) to move to the next file, :p to go back to the previous one, or :q to quit.</p>
<p>Note: If you view solutions to exercises that use less, you will see an extra command at the end that turns paging off so that we can test your solutions efficiently.</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  


  
  <nav class="pagination">
    <a class="extend prev" rel="prev noopener" href=""><i class=" target="_blank"fa fa-angle-left" aria-label="الصفحة السابقة"></i></a><a class="page-number" href="">1</a><span class=" target="_blank" rel="noopener"page-number current">2</span><a class="page-number" href="page/3/" target="_blank" rel="noopener">3</a><a class="extend next" rel="next noopener" href="page/3/" target="_blank"><i class="fa fa-angle-right" aria-label="الصفحة التالية"></i></a>
  </nav>



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          المحتويات
        </li>
        <li class="sidebar-nav-overview">
          عام
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">Yansen</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/" target="_blank" rel="noopener">
        
          <span class="site-state-item-count">25</span>
          <span class="site-state-item-name">المقالات</span>
        </a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="tags/" target="_blank" rel="noopener">
        <span class="site-state-item-count">9</span>
        <span class="site-state-item-name">الوسوم</span></a>
      </div>
  </nav>
</div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Yansen</span>
</div>
  <div class="powered-by">تطبيق الموقع <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://mist.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Mist</a>
  </div>

        








      </div>
    </footer>
  </div>

  
  <script src="lib/anime.min.js"></script>
  <script src="lib/velocity/velocity.min.js"></script>
  <script src="lib/velocity/velocity.ui.min.js"></script>

<script src="js/utils.js"></script>

<script src="js/motion.js"></script>


<script src="js/schemes/muse.js"></script>


<script src="js/next-boot.js"></script>




  















  

  

</body>
</html>
